// @refinedev/mui/dist/index.cjs
"use client"
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  AuthPage: () => AuthPage,
  AutoSaveIndicator: () => AutoSaveIndicator,
  BooleanField: () => BooleanField,
  Breadcrumb: () => Breadcrumb,
  CircularDeterminate: () => CircularDeterminate,
  CloneButton: () => CloneButton,
  Create: () => Create,
  CreateButton: () => CreateButton,
  DarkTheme: () => DarkTheme,
  DateField: () => DateField,
  DeleteButton: () => DeleteButton,
  Edit: () => Edit,
  EditButton: () => EditButton,
  EmailField: () => EmailField,
  ErrorComponent: () => ErrorComponent,
  ExportButton: () => ExportButton,
  FileField: () => FileField,
  HamburgerMenu: () => HamburgerMenu,
  Header: () => Header,
  ImportButton: () => ImportButton,
  Layout: () => Layout,
  LightTheme: () => LightTheme,
  List: () => List4,
  ListButton: () => ListButton,
  LoginPage: () => LoginPage,
  MarkdownField: () => MarkdownField,
  NotistackSnackbarContent: () => import_notistack3.SnackbarContent,
  NumberField: () => NumberField,
  ReadyPage: () => ReadyPage,
  RefineSnackbarProvider: () => RefineSnackbarProvider,
  RefineThemes: () => RefineThemes,
  RefreshButton: () => RefreshButton,
  SaveButton: () => SaveButton,
  Show: () => Show,
  ShowButton: () => ShowButton,
  Sider: () => Sider,
  SnackbarProvider: () => import_notistack3.SnackbarProvider,
  TagField: () => TagField,
  TextFieldComponent: () => TextField6,
  ThemedHeader: () => ThemedHeader,
  ThemedHeaderV2: () => ThemedHeaderV2,
  ThemedLayout: () => ThemedLayout,
  ThemedLayoutContext: () => ThemedLayoutContext,
  ThemedLayoutContextProvider: () => ThemedLayoutContextProvider,
  ThemedLayoutV2: () => ThemedLayoutV2,
  ThemedSider: () => ThemedSider,
  ThemedSiderV2: () => ThemedSiderV2,
  ThemedTitle: () => ThemedTitle,
  ThemedTitleV2: () => ThemedTitleV2,
  Title: () => Title,
  UrlField: () => UrlField,
  WelcomePage: () => WelcomePage,
  notificationProvider: () => notificationProvider,
  useAutocomplete: () => useAutocomplete,
  useDataGrid: () => useDataGrid,
  useNotificationProvider: () => useNotificationProvider,
  useSiderVisible: () => useSiderVisible,
  useSnackbar: () => import_notistack3.useSnackbar,
  useThemedLayoutContext: () => useThemedLayoutContext,
  withSnackbar: () => import_notistack3.withSnackbar
});
module.exports = __toCommonJS(src_exports);
var import_notistack3 = require("notistack");

// src/components/layout/index.tsx
var import_react3 = __toESM(require("react"));
var import_Box2 = __toESM(require("@mui/material/Box"));

// src/components/layout/sider/index.tsx
var import_react = __toESM(require("react"));
var import_Box = __toESM(require("@mui/material/Box"));
var import_Drawer = __toESM(require("@mui/material/Drawer"));
var import_List = __toESM(require("@mui/material/List"));
var import_ListItemButton = __toESM(require("@mui/material/ListItemButton"));
var import_ListItemIcon = __toESM(require("@mui/material/ListItemIcon"));
var import_ListItemText = __toESM(require("@mui/material/ListItemText"));
var import_Collapse = __toESM(require("@mui/material/Collapse"));
var import_Tooltip = __toESM(require("@mui/material/Tooltip"));
var import_Button = __toESM(require("@mui/material/Button"));
var import_IconButton = __toESM(require("@mui/material/IconButton"));
var import_ListOutlined = __toESM(require("@mui/icons-material/ListOutlined"));
var import_Logout = __toESM(require("@mui/icons-material/Logout"));
var import_ExpandLess = __toESM(require("@mui/icons-material/ExpandLess"));
var import_ExpandMore = __toESM(require("@mui/icons-material/ExpandMore"));
var import_ChevronLeft = __toESM(require("@mui/icons-material/ChevronLeft"));
var import_ChevronRight = __toESM(require("@mui/icons-material/ChevronRight"));
var import_MenuRounded = __toESM(require("@mui/icons-material/MenuRounded"));
var import_Dashboard = __toESM(require("@mui/icons-material/Dashboard"));
var import_core = require("@refinedev/core");
var Sider = /* @__PURE__ */ __name(({
  Title: TitleFromProps,
  render,
  meta
}) => {
  const [collapsed, setCollapsed] = (0, import_react.useState)(false);
  const [opened, setOpened] = (0, import_react.useState)(false);
  const drawerWidth = /* @__PURE__ */ __name(() => {
    if (collapsed)
      return 64;
    return 200;
  }, "drawerWidth");
  const t = (0, import_core.useTranslate)();
  const routerType = (0, import_core.useRouterType)();
  const Link6 = (0, import_core.useLink)();
  const { Link: LegacyLink } = (0, import_core.useRouterContext)();
  const ActiveLink = routerType === "legacy" ? LegacyLink : Link6;
  const { hasDashboard } = (0, import_core.useRefineContext)();
  const translate = (0, import_core.useTranslate)();
  const { warnWhen, setWarnWhen } = (0, import_core.useWarnAboutChange)();
  const { menuItems, selectedKey, defaultOpenKeys } = (0, import_core.useMenu)({ meta });
  const isExistAuthentication = (0, import_core.useIsExistAuthentication)();
  const TitleFromContext = (0, import_core.useTitle)();
  const authProvider = (0, import_core.useActiveAuthProvider)();
  const { mutate: mutateLogout } = (0, import_core.useLogout)({
    v3LegacyAuthProviderCompatible: Boolean(authProvider == null ? void 0 : authProvider.isLegacy)
  });
  const [open, setOpen] = (0, import_react.useState)({});
  import_react.default.useEffect(() => {
    setOpen((previous) => {
      const previousKeys = Object.keys(previous);
      const previousOpenKeys = previousKeys.filter((key) => previous[key]);
      const uniqueKeys = /* @__PURE__ */ new Set([...previousOpenKeys, ...defaultOpenKeys]);
      const uniqueKeysRecord = Object.fromEntries(
        Array.from(uniqueKeys.values()).map((key) => [key, true])
      );
      return uniqueKeysRecord;
    });
  }, [defaultOpenKeys]);
  const RenderToTitle = TitleFromProps ?? TitleFromContext ?? Title;
  const handleClick = /* @__PURE__ */ __name((key) => {
    setOpen({ ...open, [key]: !open[key] });
  }, "handleClick");
  const renderTreeView = /* @__PURE__ */ __name((tree, selectedKey2) => {
    return tree.map((item) => {
      const { icon, label, route, name, children, parentName, meta: meta2, options } = item;
      const isOpen = open[item.key || ""] || false;
      const isSelected = item.key === selectedKey2;
      const isNested = !((0, import_core.pickNotDeprecated)(meta2 == null ? void 0 : meta2.parent, options == null ? void 0 : options.parent, parentName) === void 0);
      if (children.length > 0) {
        return /* @__PURE__ */ import_react.default.createElement(
          import_core.CanAccess,
          {
            key: item.key,
            resource: name,
            action: "list",
            params: {
              resource: item
            }
          },
          /* @__PURE__ */ import_react.default.createElement("div", { key: item.key }, /* @__PURE__ */ import_react.default.createElement(
            import_Tooltip.default,
            {
              title: label ?? name,
              placement: "right",
              disableHoverListener: !collapsed,
              arrow: true
            },
            /* @__PURE__ */ import_react.default.createElement(
              import_ListItemButton.default,
              {
                onClick: () => {
                  if (collapsed) {
                    setCollapsed(false);
                    if (!isOpen) {
                      handleClick(item.key || "");
                    }
                  } else {
                    handleClick(item.key || "");
                  }
                },
                sx: {
                  pl: isNested ? 4 : 2,
                  justifyContent: "center",
                  "&.Mui-selected": {
                    "&:hover": {
                      backgroundColor: "transparent"
                    },
                    backgroundColor: "transparent"
                  }
                }
              },
              /* @__PURE__ */ import_react.default.createElement(
                import_ListItemIcon.default,
                {
                  sx: {
                    justifyContent: "center",
                    minWidth: 36,
                    color: "secondary.contrastText"
                  }
                },
                icon ?? /* @__PURE__ */ import_react.default.createElement(import_ListOutlined.default, null)
              ),
              /* @__PURE__ */ import_react.default.createElement(
                import_ListItemText.default,
                {
                  primary: label,
                  primaryTypographyProps: {
                    noWrap: true,
                    fontSize: "14px",
                    fontWeight: isSelected ? "bold" : "normal"
                  }
                }
              ),
              !collapsed && (isOpen ? /* @__PURE__ */ import_react.default.createElement(import_ExpandLess.default, null) : /* @__PURE__ */ import_react.default.createElement(import_ExpandMore.default, null))
            )
          ), !collapsed && /* @__PURE__ */ import_react.default.createElement(
            import_Collapse.default,
            {
              in: open[item.key || ""],
              timeout: "auto",
              unmountOnExit: true
            },
            /* @__PURE__ */ import_react.default.createElement(import_List.default, { component: "div", disablePadding: true }, renderTreeView(children, selectedKey2))
          ))
        );
      }
      return /* @__PURE__ */ import_react.default.createElement(
        import_core.CanAccess,
        {
          key: item.key,
          resource: name,
          action: "list",
          params: { resource: item }
        },
        /* @__PURE__ */ import_react.default.createElement(
          import_Tooltip.default,
          {
            title: label ?? name,
            placement: "right",
            disableHoverListener: !collapsed,
            arrow: true
          },
          /* @__PURE__ */ import_react.default.createElement(
            import_ListItemButton.default,
            {
              component: ActiveLink,
              to: route,
              selected: isSelected,
              onClick: () => {
                setOpened(false);
              },
              sx: {
                pl: isNested ? 4 : 2,
                py: isNested ? 1.25 : 1,
                "&.Mui-selected": {
                  "&:hover": {
                    backgroundColor: "transparent"
                  },
                  backgroundColor: "transparent"
                },
                justifyContent: "center"
              }
            },
            /* @__PURE__ */ import_react.default.createElement(
              import_ListItemIcon.default,
              {
                sx: {
                  justifyContent: "center",
                  minWidth: 36,
                  color: "secondary.contrastText"
                }
              },
              icon ?? /* @__PURE__ */ import_react.default.createElement(import_ListOutlined.default, null)
            ),
            /* @__PURE__ */ import_react.default.createElement(
              import_ListItemText.default,
              {
                primary: label,
                primaryTypographyProps: {
                  noWrap: true,
                  fontSize: "14px",
                  fontWeight: isSelected ? "bold" : "normal"
                }
              }
            )
          )
        )
      );
    });
  }, "renderTreeView");
  const dashboard = hasDashboard ? /* @__PURE__ */ import_react.default.createElement(import_core.CanAccess, { resource: "dashboard", action: "list" }, /* @__PURE__ */ import_react.default.createElement(
    import_Tooltip.default,
    {
      title: translate("dashboard.title", "Dashboard"),
      placement: "right",
      disableHoverListener: !collapsed,
      arrow: true
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_ListItemButton.default,
      {
        component: ActiveLink,
        to: "/",
        selected: selectedKey === "/",
        onClick: () => {
          setOpened(false);
        },
        sx: {
          pl: 2,
          py: 1,
          "&.Mui-selected": {
            "&:hover": {
              backgroundColor: "transparent"
            },
            backgroundColor: "transparent"
          },
          justifyContent: "center"
        }
      },
      /* @__PURE__ */ import_react.default.createElement(
        import_ListItemIcon.default,
        {
          sx: {
            justifyContent: "center",
            minWidth: 36,
            color: "secondary.contrastText"
          }
        },
        /* @__PURE__ */ import_react.default.createElement(import_Dashboard.default, null)
      ),
      /* @__PURE__ */ import_react.default.createElement(
        import_ListItemText.default,
        {
          primary: translate("dashboard.title", "Dashboard"),
          primaryTypographyProps: {
            noWrap: true,
            fontSize: "14px",
            fontWeight: selectedKey === "/" ? "bold" : "normal"
          }
        }
      )
    )
  )) : null;
  const handleLogout = /* @__PURE__ */ __name(() => {
    if (warnWhen) {
      const confirm = window.confirm(
        translate(
          "warnWhenUnsavedChanges",
          "Are you sure you want to leave? You have unsaved changes."
        )
      );
      if (confirm) {
        setWarnWhen(false);
        mutateLogout();
      }
    } else {
      mutateLogout();
    }
  }, "handleLogout");
  const logout = isExistAuthentication && /* @__PURE__ */ import_react.default.createElement(
    import_Tooltip.default,
    {
      title: t("buttons.logout", "Logout"),
      placement: "right",
      disableHoverListener: !collapsed,
      arrow: true
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_ListItemButton.default,
      {
        key: "logout",
        onClick: handleLogout,
        sx: { justifyContent: "center" }
      },
      /* @__PURE__ */ import_react.default.createElement(
        import_ListItemIcon.default,
        {
          sx: {
            justifyContent: "center",
            minWidth: 36,
            color: "secondary.contrastText"
          }
        },
        /* @__PURE__ */ import_react.default.createElement(import_Logout.default, null)
      ),
      /* @__PURE__ */ import_react.default.createElement(
        import_ListItemText.default,
        {
          primary: t("buttons.logout", "Logout"),
          primaryTypographyProps: {
            noWrap: true,
            fontSize: "14px"
          }
        }
      )
    )
  );
  const items = renderTreeView(menuItems, selectedKey);
  const renderSider = /* @__PURE__ */ __name(() => {
    if (render) {
      return render({
        dashboard,
        logout,
        items,
        collapsed
      });
    }
    return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, dashboard, items, logout);
  }, "renderSider");
  const drawer = /* @__PURE__ */ import_react.default.createElement(import_List.default, { disablePadding: true, sx: { mt: 1, color: "secondary.contrastText" } }, renderSider());
  return /* @__PURE__ */ import_react.default.createElement(import_react.default.Fragment, null, /* @__PURE__ */ import_react.default.createElement(
    import_Box.default,
    {
      sx: {
        width: { xs: drawerWidth() },
        display: {
          xs: "none",
          md: "block"
        },
        transition: "width 0.3s ease"
      }
    }
  ), /* @__PURE__ */ import_react.default.createElement(
    import_Box.default,
    {
      component: "nav",
      sx: {
        position: "fixed",
        zIndex: 1101,
        width: { sm: drawerWidth() },
        display: "flex"
      }
    },
    /* @__PURE__ */ import_react.default.createElement(
      import_Drawer.default,
      {
        variant: "temporary",
        open: opened,
        onClose: () => setOpened(false),
        ModalProps: {
          keepMounted: true
          // Better open performance on mobile.
        },
        sx: {
          display: { sm: "block", md: "none" },
          "& .MuiDrawer-paper": {
            width: 256,
            bgcolor: "secondary.main"
          }
        }
      },
      /* @__PURE__ */ import_react.default.createElement(
        import_Box.default,
        {
          sx: {
            height: 64,
            display: "flex",
            alignItems: "center",
            justifyContent: "center"
          }
        },
        /* @__PURE__ */ import_react.default.createElement(RenderToTitle, { collapsed: false })
      ),
      drawer
    ),
    /* @__PURE__ */ import_react.default.createElement(
      import_Drawer.default,
      {
        variant: "permanent",
        PaperProps: { elevation: 1 },
        sx: {
          display: { xs: "none", md: "block" },
          "& .MuiDrawer-paper": {
            width: drawerWidth,
            bgcolor: "secondary.main",
            overflow: "hidden",
            transition: "width 200ms cubic-bezier(0.4, 0, 0.6, 1) 0ms"
          }
        },
        open: true
      },
      /* @__PURE__ */ import_react.default.createElement(
        import_Box.default,
        {
          sx: {
            height: 64,
            display: "flex",
            alignItems: "center",
            justifyContent: "center"
          }
        },
        /* @__PURE__ */ import_react.default.createElement(RenderToTitle, { collapsed })
      ),
      /* @__PURE__ */ import_react.default.createElement(
        import_Box.default,
        {
          sx: {
            flexGrow: 1,
            overflowX: "hidden",
            overflowY: "auto"
          }
        },
        drawer
      ),
      /* @__PURE__ */ import_react.default.createElement(
        import_Button.default,
        {
          sx: {
            background: "rgba(0,0,0,.5)",
            color: "secondary.contrastText",
            textAlign: "center",
            borderRadius: 0,
            borderTop: "1px solid #ffffff1a"
          },
          fullWidth: true,
          size: "large",
          onClick: () => setCollapsed((prev) => !prev)
        },
        collapsed ? /* @__PURE__ */ import_react.default.createElement(import_ChevronRight.default, null) : /* @__PURE__ */ import_react.default.createElement(import_ChevronLeft.default, null)
      )
    ),
    /* @__PURE__ */ import_react.default.createElement(
      import_Box.default,
      {
        sx: {
          display: { xs: "block", md: "none" },
          position: "fixed",
          top: "64px",
          left: "0px",
          borderRadius: "0 6px 6px 0",
          bgcolor: "secondary.main",
          zIndex: 1199,
          width: "36px"
        }
      },
      /* @__PURE__ */ import_react.default.createElement(
        import_IconButton.default,
        {
          sx: { color: "#fff", width: "36px" },
          onClick: () => setOpened((prev) => !prev)
        },
        /* @__PURE__ */ import_react.default.createElement(import_MenuRounded.default, null)
      )
    )
  ));
}, "Sider");

// src/components/layout/header/index.tsx
var import_react2 = __toESM(require("react"));
var import_core2 = require("@refinedev/core");
var import_AppBar = __toESM(require("@mui/material/AppBar"));
var import_Stack = __toESM(require("@mui/material/Stack"));
var import_Toolbar = __toESM(require("@mui/material/Toolbar"));
var import_Typography = __toESM(require("@mui/material/Typography"));
var import_Avatar = __toESM(require("@mui/material/Avatar"));
var Header = /* @__PURE__ */ __name(() => {
  const authProvider = (0, import_core2.useActiveAuthProvider)();
  const { data: user } = (0, import_core2.useGetIdentity)({
    v3LegacyAuthProviderCompatible: Boolean(authProvider == null ? void 0 : authProvider.isLegacy)
  });
  const shouldRenderHeader = user && (user.name || user.avatar);
  return shouldRenderHeader ? /* @__PURE__ */ import_react2.default.createElement(import_AppBar.default, { color: "default", position: "sticky", elevation: 1 }, /* @__PURE__ */ import_react2.default.createElement(import_Toolbar.default, null, /* @__PURE__ */ import_react2.default.createElement(
    import_Stack.default,
    {
      direction: "row",
      width: "100%",
      justifyContent: "flex-end",
      alignItems: "center"
    },
    /* @__PURE__ */ import_react2.default.createElement(
      import_Stack.default,
      {
        direction: "row",
        gap: "16px",
        alignItems: "center",
        justifyContent: "center"
      },
      /* @__PURE__ */ import_react2.default.createElement(import_Typography.default, { variant: "subtitle2" }, user == null ? void 0 : user.name),
      /* @__PURE__ */ import_react2.default.createElement(import_Avatar.default, { src: user == null ? void 0 : user.avatar, alt: user == null ? void 0 : user.name })
    )
  ))) : null;
}, "Header");

// src/components/layout/index.tsx
var Layout = /* @__PURE__ */ __name(({
  Sider: Sider2,
  Header: Header2,
  Title: Title2,
  Footer,
  OffLayoutArea,
  children
}) => {
  const SiderToRender = Sider2 ?? Sider;
  const HeaderToRender = Header2 ?? Header;
  return /* @__PURE__ */ import_react3.default.createElement(import_Box2.default, { display: "flex", flexDirection: "row" }, /* @__PURE__ */ import_react3.default.createElement(SiderToRender, { Title: Title2 }), /* @__PURE__ */ import_react3.default.createElement(
    import_Box2.default,
    {
      sx: {
        display: "flex",
        flexDirection: "column",
        flex: 1,
        minHeight: "100vh"
      }
    },
    /* @__PURE__ */ import_react3.default.createElement(HeaderToRender, null),
    /* @__PURE__ */ import_react3.default.createElement(
      import_Box2.default,
      {
        component: "main",
        sx: {
          p: { xs: 1, md: 2, lg: 3 },
          flexGrow: 1,
          bgcolor: (theme) => theme.palette.background.default
        }
      },
      children
    ),
    Footer && /* @__PURE__ */ import_react3.default.createElement(Footer, null)
  ), OffLayoutArea && /* @__PURE__ */ import_react3.default.createElement(OffLayoutArea, null));
}, "Layout");

// src/components/layout/title/index.tsx
var import_react4 = __toESM(require("react"));
var import_core3 = require("@refinedev/core");
var import_Button2 = __toESM(require("@mui/material/Button"));
var Title = /* @__PURE__ */ __name(({ collapsed }) => {
  const routerType = (0, import_core3.useRouterType)();
  const Link6 = (0, import_core3.useLink)();
  const { Link: LegacyLink } = (0, import_core3.useRouterContext)();
  const ActiveLink = routerType === "legacy" ? LegacyLink : Link6;
  return /* @__PURE__ */ import_react4.default.createElement(import_Button2.default, { fullWidth: true, variant: "text", disableRipple: true }, /* @__PURE__ */ import_react4.default.createElement(ActiveLink, { to: "/" }, collapsed ? /* @__PURE__ */ import_react4.default.createElement(
    "img",
    {
      src: "https://refine.ams3.cdn.digitaloceanspaces.com/logo/refine-mini.svg",
      alt: "Refine",
      width: "28px",
      style: { maxHeight: "38px" }
    }
  ) : /* @__PURE__ */ import_react4.default.createElement(
    "img",
    {
      src: "https://refine.ams3.cdn.digitaloceanspaces.com/logo/refine.svg",
      alt: "Refine",
      width: "140px"
    }
  )));
}, "Title");

// src/components/themedLayout/index.tsx
var import_react7 = __toESM(require("react"));
var import_Box4 = __toESM(require("@mui/material/Box"));

// src/components/themedLayout/sider/index.tsx
var import_react5 = __toESM(require("react"));
var import_Box3 = __toESM(require("@mui/material/Box"));
var import_Drawer2 = __toESM(require("@mui/material/Drawer"));
var import_List2 = __toESM(require("@mui/material/List"));
var import_ListItemButton2 = __toESM(require("@mui/material/ListItemButton"));
var import_ListItemIcon2 = __toESM(require("@mui/material/ListItemIcon"));
var import_ListItemText2 = __toESM(require("@mui/material/ListItemText"));
var import_Collapse2 = __toESM(require("@mui/material/Collapse"));
var import_Tooltip2 = __toESM(require("@mui/material/Tooltip"));
var import_IconButton2 = __toESM(require("@mui/material/IconButton"));
var import_Paper = __toESM(require("@mui/material/Paper"));
var import_ListOutlined2 = __toESM(require("@mui/icons-material/ListOutlined"));
var import_Logout2 = __toESM(require("@mui/icons-material/Logout"));
var import_ExpandLess2 = __toESM(require("@mui/icons-material/ExpandLess"));
var import_ExpandMore2 = __toESM(require("@mui/icons-material/ExpandMore"));
var import_ChevronLeft2 = __toESM(require("@mui/icons-material/ChevronLeft"));
var import_MenuRounded2 = __toESM(require("@mui/icons-material/MenuRounded"));
var import_Dashboard2 = __toESM(require("@mui/icons-material/Dashboard"));
var import_core4 = require("@refinedev/core");
var ThemedSider = /* @__PURE__ */ __name(({
  Title: TitleFromProps,
  render,
  meta,
  isSiderOpen,
  onToggleSiderClick
}) => {
  const [opened, setOpened] = (0, import_react5.useState)(false);
  const drawerWidth = /* @__PURE__ */ __name(() => {
    if (isSiderOpen)
      return 240;
    return 56;
  }, "drawerWidth");
  const t = (0, import_core4.useTranslate)();
  const routerType = (0, import_core4.useRouterType)();
  const Link6 = (0, import_core4.useLink)();
  const { Link: LegacyLink } = (0, import_core4.useRouterContext)();
  const ActiveLink = routerType === "legacy" ? LegacyLink : Link6;
  const { hasDashboard } = (0, import_core4.useRefineContext)();
  const translate = (0, import_core4.useTranslate)();
  const { menuItems, selectedKey, defaultOpenKeys } = (0, import_core4.useMenu)({ meta });
  const isExistAuthentication = (0, import_core4.useIsExistAuthentication)();
  const TitleFromContext = (0, import_core4.useTitle)();
  const authProvider = (0, import_core4.useActiveAuthProvider)();
  const { warnWhen, setWarnWhen } = (0, import_core4.useWarnAboutChange)();
  const { mutate: mutateLogout } = (0, import_core4.useLogout)({
    v3LegacyAuthProviderCompatible: Boolean(authProvider == null ? void 0 : authProvider.isLegacy)
  });
  const [open, setOpen] = (0, import_react5.useState)({});
  import_react5.default.useEffect(() => {
    setOpen((previous) => {
      const previousKeys = Object.keys(previous);
      const previousOpenKeys = previousKeys.filter((key) => previous[key]);
      const uniqueKeys = /* @__PURE__ */ new Set([...previousOpenKeys, ...defaultOpenKeys]);
      const uniqueKeysRecord = Object.fromEntries(
        Array.from(uniqueKeys.values()).map((key) => [key, true])
      );
      return uniqueKeysRecord;
    });
  }, [defaultOpenKeys]);
  const RenderToTitle = TitleFromProps ?? TitleFromContext ?? ThemedTitle;
  const handleClick = /* @__PURE__ */ __name((key) => {
    setOpen({ ...open, [key]: !open[key] });
  }, "handleClick");
  const renderTreeView = /* @__PURE__ */ __name((tree, selectedKey2) => {
    return tree.map((item) => {
      const { icon, label, route, name, children, parentName, meta: meta2, options } = item;
      const isOpen = open[item.key || ""] || false;
      const isSelected = item.key === selectedKey2;
      const isNested = !((0, import_core4.pickNotDeprecated)(meta2 == null ? void 0 : meta2.parent, options == null ? void 0 : options.parent, parentName) === void 0);
      if (children.length > 0) {
        return /* @__PURE__ */ import_react5.default.createElement(
          import_core4.CanAccess,
          {
            key: item.key,
            resource: name,
            action: "list",
            params: {
              resource: item
            }
          },
          /* @__PURE__ */ import_react5.default.createElement("div", { key: item.key }, /* @__PURE__ */ import_react5.default.createElement(
            import_Tooltip2.default,
            {
              title: label ?? name,
              placement: "right",
              disableHoverListener: isSiderOpen,
              arrow: true
            },
            /* @__PURE__ */ import_react5.default.createElement(
              import_ListItemButton2.default,
              {
                onClick: () => {
                  if (!isSiderOpen) {
                    onToggleSiderClick == null ? void 0 : onToggleSiderClick(true);
                    if (!isOpen) {
                      handleClick(item.key || "");
                    }
                  } else {
                    handleClick(item.key || "");
                  }
                },
                sx: {
                  pl: isNested ? 4 : 2,
                  justifyContent: "center",
                  marginTop: "8px"
                }
              },
              /* @__PURE__ */ import_react5.default.createElement(
                import_ListItemIcon2.default,
                {
                  sx: {
                    justifyContent: "center",
                    minWidth: "24px",
                    transition: "margin-right 0.3s",
                    marginRight: isSiderOpen ? "12px" : "0px",
                    color: "currentColor"
                  }
                },
                icon ?? /* @__PURE__ */ import_react5.default.createElement(import_ListOutlined2.default, null)
              ),
              /* @__PURE__ */ import_react5.default.createElement(
                import_ListItemText2.default,
                {
                  primary: label,
                  primaryTypographyProps: {
                    noWrap: true,
                    fontSize: "14px"
                  }
                }
              ),
              isOpen ? /* @__PURE__ */ import_react5.default.createElement(
                import_ExpandLess2.default,
                {
                  sx: {
                    color: "text.icon"
                  }
                }
              ) : /* @__PURE__ */ import_react5.default.createElement(
                import_ExpandMore2.default,
                {
                  sx: {
                    color: "text.icon"
                  }
                }
              )
            )
          ), isSiderOpen && /* @__PURE__ */ import_react5.default.createElement(
            import_Collapse2.default,
            {
              in: open[item.key || ""],
              timeout: "auto",
              unmountOnExit: true
            },
            /* @__PURE__ */ import_react5.default.createElement(import_List2.default, { component: "div", disablePadding: true }, renderTreeView(children, selectedKey2))
          ))
        );
      }
      return /* @__PURE__ */ import_react5.default.createElement(
        import_core4.CanAccess,
        {
          key: item.key,
          resource: name,
          action: "list",
          params: { resource: item }
        },
        /* @__PURE__ */ import_react5.default.createElement(
          import_Tooltip2.default,
          {
            title: label ?? name,
            placement: "right",
            disableHoverListener: isSiderOpen,
            arrow: true
          },
          /* @__PURE__ */ import_react5.default.createElement(
            import_ListItemButton2.default,
            {
              component: ActiveLink,
              to: route,
              selected: isSelected,
              onClick: () => {
                setOpened(false);
              },
              sx: {
                pl: isNested ? 4 : 2,
                py: isNested ? 1.25 : 1,
                justifyContent: "center",
                color: isSelected ? "primary.main" : "text.primary"
              }
            },
            /* @__PURE__ */ import_react5.default.createElement(
              import_ListItemIcon2.default,
              {
                sx: {
                  justifyContent: "center",
                  transition: "margin-right 0.3s",
                  marginRight: isSiderOpen ? "12px" : "0px",
                  minWidth: "24px",
                  color: "currentColor"
                }
              },
              icon ?? /* @__PURE__ */ import_react5.default.createElement(import_ListOutlined2.default, null)
            ),
            /* @__PURE__ */ import_react5.default.createElement(
              import_ListItemText2.default,
              {
                primary: label,
                primaryTypographyProps: {
                  noWrap: true,
                  fontSize: "14px"
                }
              }
            )
          )
        )
      );
    });
  }, "renderTreeView");
  const dashboard = hasDashboard ? /* @__PURE__ */ import_react5.default.createElement(import_core4.CanAccess, { resource: "dashboard", action: "list" }, /* @__PURE__ */ import_react5.default.createElement(
    import_Tooltip2.default,
    {
      title: translate("dashboard.title", "Dashboard"),
      placement: "right",
      disableHoverListener: isSiderOpen,
      arrow: true
    },
    /* @__PURE__ */ import_react5.default.createElement(
      import_ListItemButton2.default,
      {
        component: ActiveLink,
        to: "/",
        selected: selectedKey === "/",
        onClick: () => {
          setOpened(false);
        },
        sx: {
          pl: 2,
          py: 1,
          justifyContent: "center",
          color: selectedKey === "/" ? "primary.main" : "text.primary"
        }
      },
      /* @__PURE__ */ import_react5.default.createElement(
        import_ListItemIcon2.default,
        {
          sx: {
            justifyContent: "center",
            minWidth: "24px",
            transition: "margin-right 0.3s",
            marginRight: isSiderOpen ? "12px" : "0px",
            color: "currentColor",
            fontSize: "14px"
          }
        },
        /* @__PURE__ */ import_react5.default.createElement(import_Dashboard2.default, null)
      ),
      /* @__PURE__ */ import_react5.default.createElement(
        import_ListItemText2.default,
        {
          primary: translate("dashboard.title", "Dashboard"),
          primaryTypographyProps: {
            noWrap: true,
            fontSize: "14px"
          }
        }
      )
    )
  )) : null;
  const handleLogout = /* @__PURE__ */ __name(() => {
    if (warnWhen) {
      const confirm = window.confirm(
        t(
          "warnWhenUnsavedChanges",
          "Are you sure you want to leave? You have unsaved changes."
        )
      );
      if (confirm) {
        setWarnWhen(false);
        mutateLogout();
      }
    } else {
      mutateLogout();
    }
  }, "handleLogout");
  const logout = isExistAuthentication && /* @__PURE__ */ import_react5.default.createElement(
    import_Tooltip2.default,
    {
      title: t("buttons.logout", "Logout"),
      placement: "right",
      disableHoverListener: isSiderOpen,
      arrow: true
    },
    /* @__PURE__ */ import_react5.default.createElement(
      import_ListItemButton2.default,
      {
        key: "logout",
        onClick: () => handleLogout(),
        sx: {
          justifyContent: "center"
        }
      },
      /* @__PURE__ */ import_react5.default.createElement(
        import_ListItemIcon2.default,
        {
          sx: {
            justifyContent: "center",
            minWidth: "24px",
            transition: "margin-right 0.3s",
            marginRight: isSiderOpen ? "12px" : "0px",
            marginLeft: "2px",
            color: "currentColor"
          }
        },
        /* @__PURE__ */ import_react5.default.createElement(import_Logout2.default, null)
      ),
      /* @__PURE__ */ import_react5.default.createElement(
        import_ListItemText2.default,
        {
          primary: t("buttons.logout", "Logout"),
          primaryTypographyProps: {
            noWrap: true,
            fontSize: "14px"
          }
        }
      )
    )
  );
  const items = renderTreeView(menuItems, selectedKey);
  const renderSider = /* @__PURE__ */ __name(() => {
    if (render) {
      return render({
        dashboard,
        logout,
        items,
        collapsed: !isSiderOpen
      });
    }
    return /* @__PURE__ */ import_react5.default.createElement(import_react5.default.Fragment, null, dashboard, items, logout);
  }, "renderSider");
  const drawer = /* @__PURE__ */ import_react5.default.createElement(
    import_List2.default,
    {
      disablePadding: true,
      sx: {
        flexGrow: 1,
        paddingTop: "16px"
      }
    },
    renderSider()
  );
  return /* @__PURE__ */ import_react5.default.createElement(import_react5.default.Fragment, null, /* @__PURE__ */ import_react5.default.createElement(
    import_Box3.default,
    {
      sx: {
        width: { xs: drawerWidth() },
        display: {
          xs: "none",
          md: "block"
        },
        transition: "width 0.3s ease"
      }
    }
  ), /* @__PURE__ */ import_react5.default.createElement(
    import_Box3.default,
    {
      component: "nav",
      sx: {
        position: "fixed",
        zIndex: 1101,
        width: { sm: drawerWidth() },
        display: "flex"
      }
    },
    /* @__PURE__ */ import_react5.default.createElement(
      import_Drawer2.default,
      {
        variant: "temporary",
        elevation: 2,
        open: opened,
        onClose: () => setOpened(false),
        ModalProps: {
          keepMounted: true
          // Better open performance on mobile.
        },
        sx: {
          display: {
            sm: "block",
            md: "none"
          }
        }
      },
      /* @__PURE__ */ import_react5.default.createElement(
        import_Box3.default,
        {
          sx: {
            width: drawerWidth()
          }
        },
        /* @__PURE__ */ import_react5.default.createElement(
          import_Box3.default,
          {
            sx: {
              height: 64,
              display: "flex",
              alignItems: "center",
              paddingLeft: "16px",
              fontSize: "14px"
            }
          },
          /* @__PURE__ */ import_react5.default.createElement(RenderToTitle, { collapsed: false })
        ),
        drawer
      )
    ),
    /* @__PURE__ */ import_react5.default.createElement(
      import_Drawer2.default,
      {
        variant: "permanent",
        sx: {
          display: { xs: "none", md: "block" },
          "& .MuiDrawer-paper": {
            width: drawerWidth,
            overflow: "hidden",
            transition: "width 200ms cubic-bezier(0.4, 0, 0.6, 1) 0ms"
          }
        },
        open: true
      },
      /* @__PURE__ */ import_react5.default.createElement(
        import_Paper.default,
        {
          elevation: 0,
          sx: {
            fontSize: "14px",
            width: "100%",
            height: 64,
            display: "flex",
            alignItems: "center",
            justifyContent: !isSiderOpen ? "center" : "space-between",
            paddingLeft: !isSiderOpen ? 0 : "16px",
            paddingRight: !isSiderOpen ? 0 : "8px",
            variant: "outlined",
            borderRadius: 0,
            borderBottom: (theme) => `1px solid ${theme.palette.action.focus}`
          }
        },
        /* @__PURE__ */ import_react5.default.createElement(RenderToTitle, { collapsed: !isSiderOpen }),
        isSiderOpen && /* @__PURE__ */ import_react5.default.createElement(
          import_IconButton2.default,
          {
            size: "small",
            onClick: () => onToggleSiderClick == null ? void 0 : onToggleSiderClick(false)
          },
          /* @__PURE__ */ import_react5.default.createElement(import_ChevronLeft2.default, null)
        )
      ),
      /* @__PURE__ */ import_react5.default.createElement(
        import_Box3.default,
        {
          sx: {
            flexGrow: 1,
            overflowX: "hidden",
            overflowY: "auto"
          }
        },
        drawer
      )
    ),
    /* @__PURE__ */ import_react5.default.createElement(
      import_IconButton2.default,
      {
        sx: (theme) => {
          const { palette } = theme;
          const { mode, getContrastText, primary, background } = palette;
          return {
            color: mode === "light" ? getContrastText(primary.main) : getContrastText(background.paper),
            display: { xs: "flex", md: "none" },
            justifyContent: "center",
            alignItems: "center",
            position: "fixed",
            top: "12px",
            left: "16px",
            zIndex: 1199,
            height: "36px",
            width: "36px"
          };
        },
        onClick: () => setOpened((prev) => !prev)
      },
      /* @__PURE__ */ import_react5.default.createElement(import_MenuRounded2.default, null)
    )
  ));
}, "ThemedSider");

// src/components/themedLayout/header/index.tsx
var import_react6 = __toESM(require("react"));
var import_core5 = require("@refinedev/core");
var import_AppBar2 = __toESM(require("@mui/material/AppBar"));
var import_Stack2 = __toESM(require("@mui/material/Stack"));
var import_Toolbar2 = __toESM(require("@mui/material/Toolbar"));
var import_Typography2 = __toESM(require("@mui/material/Typography"));
var import_Avatar2 = __toESM(require("@mui/material/Avatar"));
var import_IconButton3 = __toESM(require("@mui/material/IconButton"));
var import_Menu = __toESM(require("@mui/icons-material/Menu"));
var ThemedHeader = /* @__PURE__ */ __name(({
  isSiderOpen,
  onToggleSiderClick,
  toggleSiderIcon: toggleSiderIconFromProps
}) => {
  const authProvider = (0, import_core5.useActiveAuthProvider)();
  const { data: user } = (0, import_core5.useGetIdentity)({
    v3LegacyAuthProviderCompatible: Boolean(authProvider == null ? void 0 : authProvider.isLegacy)
  });
  const hasSidebarToggle = Boolean(onToggleSiderClick);
  return /* @__PURE__ */ import_react6.default.createElement(import_AppBar2.default, { position: "sticky" }, /* @__PURE__ */ import_react6.default.createElement(import_Toolbar2.default, null, hasSidebarToggle && /* @__PURE__ */ import_react6.default.createElement(
    import_IconButton3.default,
    {
      color: "inherit",
      "aria-label": "open drawer",
      onClick: () => onToggleSiderClick == null ? void 0 : onToggleSiderClick(),
      edge: "start",
      sx: {
        mr: 2,
        display: { xs: "none", md: "flex" },
        ...isSiderOpen && { display: "none" }
      }
    },
    (toggleSiderIconFromProps == null ? void 0 : toggleSiderIconFromProps(Boolean(isSiderOpen))) ?? /* @__PURE__ */ import_react6.default.createElement(import_Menu.default, null)
  ), /* @__PURE__ */ import_react6.default.createElement(
    import_Stack2.default,
    {
      direction: "row",
      width: "100%",
      justifyContent: "flex-end",
      alignItems: "center"
    },
    /* @__PURE__ */ import_react6.default.createElement(
      import_Stack2.default,
      {
        direction: "row",
        gap: "16px",
        alignItems: "center",
        justifyContent: "center"
      },
      /* @__PURE__ */ import_react6.default.createElement(
        import_Typography2.default,
        {
          sx: {
            display: { xs: "none", md: "block" }
          },
          variant: "subtitle2"
        },
        user == null ? void 0 : user.name
      ),
      /* @__PURE__ */ import_react6.default.createElement(import_Avatar2.default, { src: user == null ? void 0 : user.avatar, alt: user == null ? void 0 : user.name })
    )
  )));
}, "ThemedHeader");

// src/components/themedLayout/index.tsx
var ThemedLayout = /* @__PURE__ */ __name(({
  Sider: Sider2,
  Header: Header2,
  Title: Title2,
  Footer,
  OffLayoutArea,
  children
}) => {
  const [isSiderOpen, setIsSiderOpen] = (0, import_react7.useState)(true);
  const SiderToRender = Sider2 ?? ThemedSider;
  const HeaderToRender = Header2 ?? ThemedHeader;
  return /* @__PURE__ */ import_react7.default.createElement(import_Box4.default, { display: "flex", flexDirection: "row" }, /* @__PURE__ */ import_react7.default.createElement(
    SiderToRender,
    {
      Title: Title2,
      isSiderOpen,
      onToggleSiderClick: (isOpen) => setIsSiderOpen(Boolean(isOpen))
    }
  ), /* @__PURE__ */ import_react7.default.createElement(
    import_Box4.default,
    {
      sx: {
        display: "flex",
        flexDirection: "column",
        flex: 1,
        minHeight: "100vh"
      }
    },
    /* @__PURE__ */ import_react7.default.createElement(
      HeaderToRender,
      {
        isSiderOpen,
        onToggleSiderClick: () => setIsSiderOpen((prev) => !prev)
      }
    ),
    /* @__PURE__ */ import_react7.default.createElement(
      import_Box4.default,
      {
        component: "main",
        sx: {
          p: { xs: 1, md: 2, lg: 3 },
          flexGrow: 1,
          bgcolor: (theme) => theme.palette.background.default
        }
      },
      children
    ),
    Footer && /* @__PURE__ */ import_react7.default.createElement(Footer, null)
  ), OffLayoutArea && /* @__PURE__ */ import_react7.default.createElement(OffLayoutArea, null));
}, "ThemedLayout");

// src/components/themedLayout/title/index.tsx
var import_react8 = __toESM(require("react"));
var import_core6 = require("@refinedev/core");
var import_Link = __toESM(require("@mui/material/Link"));
var import_SvgIcon = __toESM(require("@mui/material/SvgIcon"));
var import_Typography3 = __toESM(require("@mui/material/Typography"));
var defaultText = "Refine Project";
var defaultIcon = /* @__PURE__ */ import_react8.default.createElement(
  "svg",
  {
    width: 24,
    height: 24,
    viewBox: "0 0 24 24",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg",
      },
  /* @__PURE__ */ import_react8.default.createElement(
    "path",
    {
      fillRule: "evenodd",
      clipRule: "evenodd",
      d: "M13.7889 0.422291C12.6627 -0.140764 11.3373 -0.140764 10.2111 0.422291L2.21115 4.42229C0.85601 5.09986 0 6.48491 0 8V16C0 17.5151 0.85601 18.9001 2.21115 19.5777L10.2111 23.5777C11.3373 24.1408 12.6627 24.1408 13.7889 23.5777L21.7889 19.5777C23.144 18.9001 24 17.5151 24 16V8C24 6.48491 23.144 5.09986 21.7889 4.42229L13.7889 0.422291ZM8 8C8 5.79086 9.79086 4 12 4C14.2091 4 16 5.79086 16 8V16C16 18.2091 14.2091 20 12 20C9.79086 20 8 18.2091 8 16V8Z",
      fill: "currentColor"
    }
  ),
  /* @__PURE__ */ import_react8.default.createElement(
    "path",
    {
      d: "M14 8C14 9.10457 13.1046 10 12 10C10.8954 10 10 9.10457 10 8C10 6.89543 10.8954 6 12 6C13.1046 6 14 6.89543 14 8Z",
      fill: "currentColor"
    }
  )
);
var ThemedTitle = /* @__PURE__ */ __name(({
  collapsed,
  wrapperStyles,
  icon = defaultIcon,
  text = defaultText
}) => {
  const routerType = (0, import_core6.useRouterType)();
  const Link6 = (0, import_core6.useLink)();
  const { Link: LegacyLink } = (0, import_core6.useRouterContext)();
  const ActiveLink = routerType === "legacy" ? LegacyLink : Link6;
  return /* @__PURE__ */ import_react8.default.createElement(
    import_Link.default,
    {
      to: "/",
      component: ActiveLink,
      underline: "none",
      sx: {
        display: "flex",
        alignItems: "center",
        gap: "12px",
        ...wrapperStyles
      }
    },
    /* @__PURE__ */ import_react8.default.createElement(import_SvgIcon.default, { height: "24px", width: "24px", color: "primary" }, icon),
    !collapsed && /* @__PURE__ */ import_react8.default.createElement(
      import_Typography3.default,
      {
        variant: "h6",
        fontWeight: 700,
        color: "text.primary",
        fontSize: "inherit",
        textOverflow: "ellipsis",
        overflow: "hidden"
      },
      text
    )
  );
}, "ThemedTitle");

// src/components/themedLayoutV2/index.tsx
var import_react16 = __toESM(require("react"));
var import_Box6 = __toESM(require("@mui/material/Box"));

// src/contexts/themedLayoutContext/index.tsx
var import_react9 = __toESM(require("react"));
var ThemedLayoutContext = import_react9.default.createContext({
  siderCollapsed: false,
  mobileSiderOpen: false,
  setSiderCollapsed: () => void 0,
  setMobileSiderOpen: () => void 0
});
var ThemedLayoutContextProvider = /* @__PURE__ */ __name(({ children, initialSiderCollapsed }) => {
  const [siderCollapsed, setSiderCollapsed] = (0, import_react9.useState)(
    initialSiderCollapsed ?? false
  );
  const [mobileSiderOpen, setMobileSiderOpen] = (0, import_react9.useState)(false);
  return /* @__PURE__ */ import_react9.default.createElement(
    ThemedLayoutContext.Provider,
    {
      value: {
        siderCollapsed,
        mobileSiderOpen,
        setSiderCollapsed,
        setMobileSiderOpen
      }
    },
    children
  );
}, "ThemedLayoutContextProvider");

// src/components/themedLayoutV2/sider/index.tsx
var import_react13 = __toESM(require("react"));
var import_Box5 = __toESM(require("@mui/material/Box"));
var import_Drawer3 = __toESM(require("@mui/material/Drawer"));
var import_List3 = __toESM(require("@mui/material/List"));
var import_ListItemButton3 = __toESM(require("@mui/material/ListItemButton"));
var import_ListItemIcon3 = __toESM(require("@mui/material/ListItemIcon"));
var import_ListItemText3 = __toESM(require("@mui/material/ListItemText"));
var import_Collapse3 = __toESM(require("@mui/material/Collapse"));
var import_Tooltip3 = __toESM(require("@mui/material/Tooltip"));
var import_IconButton4 = __toESM(require("@mui/material/IconButton"));
var import_Paper2 = __toESM(require("@mui/material/Paper"));
var import_ListOutlined3 = __toESM(require("@mui/icons-material/ListOutlined"));
var import_Logout3 = __toESM(require("@mui/icons-material/Logout"));
var import_ExpandLess3 = __toESM(require("@mui/icons-material/ExpandLess"));
var import_ExpandMore3 = __toESM(require("@mui/icons-material/ExpandMore"));
var import_ChevronLeft3 = __toESM(require("@mui/icons-material/ChevronLeft"));
var import_Dashboard3 = __toESM(require("@mui/icons-material/Dashboard"));
var import_core9 = require("@refinedev/core");

// src/hooks/useAutocomplete/index.ts
var import_core7 = require("@refinedev/core");
var import_isEqual = __toESM(require("lodash/isEqual"));
var import_unionWith = __toESM(require("lodash/unionWith"));
var useAutocomplete = /* @__PURE__ */ __name((props) => {
  var _a, _b, _c, _d;
  const { query, defaultValueQuery, onSearch, overtime } = (0, import_core7.useSelect)(props);
  return {
    autocompleteProps: {
      options: props.selectedOptionsOrder === "selected-first" ? (0, import_unionWith.default)(
        ((_a = defaultValueQuery.data) == null ? void 0 : _a.data) || [],
        ((_b = query.data) == null ? void 0 : _b.data) || [],
        import_isEqual.default
      ) : (0, import_unionWith.default)(
        ((_c = query.data) == null ? void 0 : _c.data) || [],
        ((_d = defaultValueQuery.data) == null ? void 0 : _d.data) || [],
        import_isEqual.default
      ),
      loading: query.isFetching || defaultValueQuery.isFetching,
      onInputChange: (event, value) => {
        if ((event == null ? void 0 : event.type) === "change") {
          onSearch(value);
        } else if ((event == null ? void 0 : event.type) === "click") {
          onSearch("");
        }
      },
      filterOptions: (x) => x
    },
    onSearch,
    query,
    defaultValueQuery,
    queryResult: query,
    defaultValueQueryResult: defaultValueQuery,
    overtime
  };
}, "useAutocomplete");

// src/hooks/useDataGrid/index.ts
var import_core8 = require("@refinedev/core");
var import_react10 = require("react");
var import_styles = require("@mui/material/styles");
var import_differenceWith = __toESM(require("lodash/differenceWith"));
var import_isEqual2 = __toESM(require("lodash/isEqual"));

// src/definitions/dataGrid/index.ts
var import_x_data_grid = require("@mui/x-data-grid");
var transformSortModelToCrudSorting = /* @__PURE__ */ __name((sortModel) => {
  const sorter = sortModel.map(({ field, sort }) => ({
    field,
    order: sort || "asc"
  }));
  return sorter;
}, "transformSortModelToCrudSorting");
var transformCrudSortingToSortModel = /* @__PURE__ */ __name((crudSorting) => {
  const sortModel = crudSorting.map(({ field, order }) => ({
    field,
    sort: order
  }));
  return sortModel;
}, "transformCrudSortingToSortModel");
var transformMuiOperatorToCrudOperator = /* @__PURE__ */ __name((operatorValue) => {
  if (!operatorValue) {
    return "eq";
  }
  switch (operatorValue) {
    case "equals":
    case "is":
    case "=":
      return "eq";
    case "!=":
    case "not":
      return "ne";
    case "contains":
      return "contains";
    case "isAnyOf":
      return "in";
    case ">":
    case "after":
      return "gt";
    case ">=":
    case "onOrAfter":
      return "gte";
    case "<":
    case "before":
      return "lt";
    case "<=":
    case "onOrBefore":
      return "lte";
    case "startsWith":
      return "startswith";
    case "endsWith":
      return "endswith";
    case "isEmpty":
      return "null";
    case "isNotEmpty":
      return "nnull";
    default:
      return operatorValue;
  }
}, "transformMuiOperatorToCrudOperator");
var transformFilterModelToCrudFilters = /* @__PURE__ */ __name(({
  items,
  logicOperator
}) => {
  const filters = items.map(({ field, value, operator }) => {
    const filter = {
      field,
      value: ["isEmpty", "isNotEmpty"].includes(operator) ? true : value ?? "",
      operator: transformMuiOperatorToCrudOperator(operator)
    };
    return filter;
  });
  if (logicOperator === import_x_data_grid.GridLogicOperator.Or) {
    return [{ operator: "or", value: filters }];
  }
  return filters;
}, "transformFilterModelToCrudFilters");
var transformCrudOperatorToMuiOperator = /* @__PURE__ */ __name((operator, columnType) => {
  switch (columnType) {
    case "number":
      switch (operator) {
        case "eq":
          return "=";
        case "ne":
          return "!=";
        case "gt":
          return ">";
        case "gte":
          return ">=";
        case "lt":
          return "<";
        case "lte":
          return "<=";
        case "null":
          return "isEmpty";
        case "nnull":
          return "isNotEmpty";
        case "in":
          return "isAnyOf";
        default:
          return operator;
      }
    case "singleSelect":
    case "boolean":
      switch (operator) {
        case "eq":
          return "is";
        default:
          return operator;
      }
    case void 0:
    case "string":
      switch (operator) {
        case "eq":
          return "equals";
        case "contains":
          return "contains";
        case "null":
          return "isEmpty";
        case "nnull":
          return "isNotEmpty";
        case "startswith":
          return "startsWith";
        case "endswith":
          return "endsWith";
        case "in":
          return "isAnyOf";
        default:
          return operator;
      }
    case "date":
    case "dateTime":
      switch (operator) {
        case "eq":
          return "is";
        case "ne":
          return "not";
        case "gt":
          return "after";
        case "gte":
          return "onOrAfter";
        case "lt":
          return "before";
        case "lte":
          return "onOrBefore";
        case "null":
          return "isEmpty";
        case "nnull":
          return "isNotEmpty";
        default:
          return operator;
      }
    default:
      return operator;
  }
}, "transformCrudOperatorToMuiOperator");
var transformCrudFiltersToFilterModel = /* @__PURE__ */ __name((crudFilters, columnsType) => {
  var _a;
  const gridFilterItems = [];
  const isExistOrFilter = crudFilters.some(
    (filter) => filter.operator === "or"
  );
  if (columnsType) {
    if (isExistOrFilter) {
      const orLogicalFilters = (_a = crudFilters.find(
        (filter) => filter.operator === "or"
      )) == null ? void 0 : _a.value;
      orLogicalFilters.map(({ field, value, operator }) => {
        const columnType = columnsType[field];
        gridFilterItems.push({
          field,
          operator: transformCrudOperatorToMuiOperator(operator, columnType),
          value: value === "" ? void 0 : value,
          id: field + operator
        });
      });
    } else {
      crudFilters.map(({ field, value, operator }) => {
        const columnType = columnsType[field];
        gridFilterItems.push({
          field,
          operator: transformCrudOperatorToMuiOperator(operator, columnType),
          value: value === "" ? void 0 : value,
          id: field + operator
        });
      });
    }
  }
  return {
    items: gridFilterItems,
    // If there is "or" filter, default link operator is "or"
    logicOperator: isExistOrFilter ? import_x_data_grid.GridLogicOperator.Or : import_x_data_grid.GridLogicOperator.And
  };
}, "transformCrudFiltersToFilterModel");

// src/hooks/useDataGrid/index.ts
function useDataGrid({
  onSearch: onSearchProp,
  initialCurrent,
  initialPageSize = 25,
  pagination,
  hasPagination = true,
  initialSorter,
  permanentSorter,
  defaultSetFilterBehavior = "replace",
  initialFilter,
  permanentFilter,
  filters: filtersFromProp,
  sorters: sortersFromProp,
  syncWithLocation: syncWithLocationProp,
  resource: resourceFromProp,
  successNotification,
  errorNotification,
  queryOptions,
  liveMode: liveModeFromProp,
  onLiveEvent,
  liveParams,
  meta,
  metaData,
  dataProviderName,
  overtimeOptions,
  editable = false,
  updateMutationOptions
} = {}) {
  const theme = (0, import_styles.useTheme)();
  const liveMode = (0, import_core8.useLiveMode)(liveModeFromProp);
  const [columnsTypes, setColumnsType] = (0, import_react10.useState)();
  const { identifier } = (0, import_core8.useResourceParams)({ resource: resourceFromProp });
  const {
    tableQueryResult,
    tableQuery,
    current,
    setCurrent,
    pageSize,
    setPageSize,
    filters,
    setFilters,
    sorters,
    setSorters,
    sorter,
    setSorter,
    pageCount,
    createLinkForSyncWithLocation,
    overtime
  } = (0, import_core8.useTable)({
    permanentSorter,
    permanentFilter,
    initialCurrent,
    initialPageSize,
    pagination,
    hasPagination,
    initialSorter,
    initialFilter,
    filters: filtersFromProp,
    sorters: sortersFromProp,
    syncWithLocation: syncWithLocationProp,
    defaultSetFilterBehavior,
    resource: resourceFromProp,
    successNotification,
    errorNotification,
    queryOptions,
    liveMode: liveModeFromProp,
    onLiveEvent,
    liveParams,
    meta: (0, import_core8.pickNotDeprecated)(meta, metaData),
    metaData: (0, import_core8.pickNotDeprecated)(meta, metaData),
    dataProviderName,
    overtimeOptions
  });
  const [muiCrudFilters, setMuiCrudFilters] = (0, import_react10.useState)(filters);
  const { data, isFetched, isLoading } = tableQueryResult;
  const isServerSideFilteringEnabled = ((filtersFromProp == null ? void 0 : filtersFromProp.mode) || "server") === "server";
  const isServerSideSortingEnabled = ((sortersFromProp == null ? void 0 : sortersFromProp.mode) || "server") === "server";
  const hasPaginationString = hasPagination === false ? "off" : "server";
  const isPaginationEnabled = ((pagination == null ? void 0 : pagination.mode) ?? hasPaginationString) !== "off";
  const preferredPermanentSorters = (0, import_core8.pickNotDeprecated)(sortersFromProp == null ? void 0 : sortersFromProp.permanent, permanentSorter) ?? [];
  const preferredPermanentFilters = (0, import_core8.pickNotDeprecated)(filtersFromProp == null ? void 0 : filtersFromProp.permanent, permanentFilter) ?? [];
  const handlePageChange = /* @__PURE__ */ __name((page) => {
    if (isPaginationEnabled) {
      setCurrent(page + 1);
    }
  }, "handlePageChange");
  const handlePageSizeChange = /* @__PURE__ */ __name((pageSize2) => {
    if (isPaginationEnabled) {
      setPageSize(pageSize2);
    }
  }, "handlePageSizeChange");
  const handleSortModelChange = /* @__PURE__ */ __name((sortModel) => {
    const crudSorting = transformSortModelToCrudSorting(sortModel);
    setSorters(crudSorting);
  }, "handleSortModelChange");
  const handleFilterModelChange = /* @__PURE__ */ __name((filterModel) => {
    const crudFilters = transformFilterModelToCrudFilters(filterModel);
    setMuiCrudFilters(crudFilters);
    setFilters(crudFilters.filter((f) => f.value !== ""));
    if (isPaginationEnabled) {
      setCurrent(1);
    }
  }, "handleFilterModelChange");
  const search = /* @__PURE__ */ __name(async (value) => {
    if (onSearchProp) {
      const searchFilters = await onSearchProp(value);
      setMuiCrudFilters(searchFilters);
      setFilters(searchFilters.filter((f) => f.value !== ""));
      if (isPaginationEnabled) {
        setCurrent(1);
      }
    }
  }, "search");
  const dataGridPaginationValues = /* @__PURE__ */ __name(() => {
    if (isPaginationEnabled) {
      return {
        paginationMode: "server",
        paginationModel: {
          page: current - 1,
          pageSize
        },
        onPaginationModelChange: (model) => {
          handlePageChange(model.page);
          handlePageSizeChange(model.pageSize);
        }
      };
    }
    return {
      paginationMode: "client"
    };
  }, "dataGridPaginationValues");
  const { mutate } = (0, import_core8.useUpdate)({
    mutationOptions: updateMutationOptions
  });
  const processRowUpdate = /* @__PURE__ */ __name(async (newRow, oldRow) => {
    if (!editable) {
      return Promise.resolve(oldRow);
    }
    if (!identifier) {
      return Promise.reject(new Error("Resource is not defined"));
    }
    return new Promise((resolve, reject) => {
      mutate(
        {
          resource: identifier,
          id: newRow.id,
          values: newRow
        },
        {
          onError: (error) => {
            reject(error);
          },
          onSuccess: (data2) => {
            resolve(newRow);
          }
        }
      );
    });
  }, "processRowUpdate");
  return {
    tableQueryResult,
    tableQuery,
    dataGridProps: {
      disableRowSelectionOnClick: true,
      rows: (data == null ? void 0 : data.data) || [],
      loading: liveMode === "auto" ? isLoading : !isFetched,
      rowCount: (data == null ? void 0 : data.total) || 0,
      ...dataGridPaginationValues(),
      sortingMode: isServerSideSortingEnabled ? "server" : "client",
      sortModel: transformCrudSortingToSortModel(
        (0, import_differenceWith.default)(sorters, preferredPermanentSorters, import_isEqual2.default)
      ),
      onSortModelChange: handleSortModelChange,
      filterMode: isServerSideFilteringEnabled ? "server" : "client",
      filterModel: transformCrudFiltersToFilterModel(
        (0, import_differenceWith.default)(muiCrudFilters, preferredPermanentFilters, import_isEqual2.default),
        columnsTypes
      ),
      onFilterModelChange: handleFilterModelChange,
      onStateChange: (state) => {
        const newColumnsTypes = Object.fromEntries(
          Object.entries(state.columns.lookup).map(([key, value]) => {
            return [key, value.type];
          })
        );
        const isStateChanged = !(0, import_isEqual2.default)(newColumnsTypes, columnsTypes);
        if (isStateChanged) {
          setColumnsType(newColumnsTypes);
        }
      },
      sx: {
        border: "none",
        "& .MuiDataGrid-columnHeaders": {
          background: (0, import_styles.darken)(theme.palette.background.paper, 0.05),
          borderBottom: `1px solid ${(0, import_styles.darken)(
            theme.palette.background.paper,
            0.1
          )}`
        },
        "& .MuiDataGrid-cell": {
          borderBottom: `1px solid ${(0, import_styles.darken)(
            theme.palette.background.paper,
            0.05
          )}`
        }
      },
      processRowUpdate: editable ? processRowUpdate : void 0
    },
    current,
    setCurrent,
    pageSize,
    setPageSize,
    pageCount,
    sorters,
    setSorters,
    sorter,
    setSorter,
    filters,
    setFilters,
    search,
    createLinkForSyncWithLocation,
    overtime
  };
}
__name(useDataGrid, "useDataGrid");

// src/hooks/useSiderVisible/index.ts
var import_react11 = require("react");
var useSiderVisible = /* @__PURE__ */ __name(() => {
  const {
    mobileSiderOpen,
    siderCollapsed,
    setMobileSiderOpen,
    setSiderCollapsed
  } = (0, import_react11.useContext)(ThemedLayoutContext);
  return {
    siderVisible: mobileSiderOpen,
    setSiderVisible: setMobileSiderOpen,
    drawerSiderVisible: !siderCollapsed,
    setDrawerSiderVisible: (visible) => setSiderCollapsed(!visible)
  };
}, "useSiderVisible");

// src/hooks/useThemedLayoutContext/index.ts
var import_react12 = require("react");
var useThemedLayoutContext = /* @__PURE__ */ __name(() => {
  const {
    mobileSiderOpen,
    siderCollapsed,
    setMobileSiderOpen,
    setSiderCollapsed
  } = (0, import_react12.useContext)(ThemedLayoutContext);
  return {
    mobileSiderOpen,
    siderCollapsed,
    setMobileSiderOpen,
    setSiderCollapsed
  };
}, "useThemedLayoutContext");

// src/components/themedLayoutV2/sider/index.tsx
var ThemedSiderV2 = /* @__PURE__ */ __name(({
  Title: TitleFromProps,
  render,
  meta,
  activeItemDisabled = false
}) => {
  const {
    siderCollapsed,
    setSiderCollapsed,
    mobileSiderOpen,
    setMobileSiderOpen
  } = useThemedLayoutContext();
  const drawerWidth = /* @__PURE__ */ __name(() => {
    if (siderCollapsed)
      return 56;
    return 240;
  }, "drawerWidth");
  const t = (0, import_core9.useTranslate)();
  const routerType = (0, import_core9.useRouterType)();
  const Link6 = (0, import_core9.useLink)();
  const { Link: LegacyLink } = (0, import_core9.useRouterContext)();
  const ActiveLink = routerType === "legacy" ? LegacyLink : Link6;
  const { hasDashboard } = (0, import_core9.useRefineContext)();
  const translate = (0, import_core9.useTranslate)();
  const { menuItems, selectedKey, defaultOpenKeys } = (0, import_core9.useMenu)({ meta });
  const isExistAuthentication = (0, import_core9.useIsExistAuthentication)();
  const TitleFromContext = (0, import_core9.useTitle)();
  const authProvider = (0, import_core9.useActiveAuthProvider)();
  const { warnWhen, setWarnWhen } = (0, import_core9.useWarnAboutChange)();
  const { mutate: mutateLogout } = (0, import_core9.useLogout)({
    v3LegacyAuthProviderCompatible: Boolean(authProvider == null ? void 0 : authProvider.isLegacy)
  });
  const [open, setOpen] = (0, import_react13.useState)({});
  import_react13.default.useEffect(() => {
    setOpen((previous) => {
      const previousKeys = Object.keys(previous);
      const previousOpenKeys = previousKeys.filter((key) => previous[key]);
      const uniqueKeys = /* @__PURE__ */ new Set([...previousOpenKeys, ...defaultOpenKeys]);
      const uniqueKeysRecord = Object.fromEntries(
        Array.from(uniqueKeys.values()).map((key) => [key, true])
      );
      return uniqueKeysRecord;
    });
  }, [defaultOpenKeys]);
  const RenderToTitle = TitleFromProps ?? TitleFromContext ?? ThemedTitleV2;
  const handleClick = /* @__PURE__ */ __name((key) => {
    setOpen({ ...open, [key]: !open[key] });
  }, "handleClick");
  const renderTreeView = /* @__PURE__ */ __name((tree, selectedKey2) => {
    return tree.map((item) => {
      const { icon, label, route, name, children, parentName, meta: meta2, options } = item;
      const isOpen = open[item.key || ""] || false;
      const isSelected = item.key === selectedKey2;
      const isNested = !((0, import_core9.pickNotDeprecated)(meta2 == null ? void 0 : meta2.parent, options == null ? void 0 : options.parent, parentName) === void 0);
      if (children.length > 0) {
        return /* @__PURE__ */ import_react13.default.createElement(
          import_core9.CanAccess,
          {
            key: item.key,
            resource: name,
            action: "list",
            params: {
              resource: item
            }
          },
          /* @__PURE__ */ import_react13.default.createElement("div", { key: item.key }, /* @__PURE__ */ import_react13.default.createElement(
            import_Tooltip3.default,
            {
              title: label ?? name,
              placement: "right",
              disableHoverListener: !siderCollapsed,
              arrow: true
            },
            /* @__PURE__ */ import_react13.default.createElement(
              import_ListItemButton3.default,
              {
                onClick: () => {
                  if (siderCollapsed) {
                    setSiderCollapsed(false);
                    if (!isOpen) {
                      handleClick(item.key || "");
                    }
                  } else {
                    handleClick(item.key || "");
                  }
                },
                sx: {
                  pl: isNested ? 4 : 2,
                  justifyContent: "center"
                }
              },
              /* @__PURE__ */ import_react13.default.createElement(
                import_ListItemIcon3.default,
                {
                  sx: {
                    justifyContent: "center",
                    minWidth: "24px",
                    transition: "margin-right 0.3s",
                    marginRight: siderCollapsed ? "0px" : "12px",
                    color: "currentColor"
                  }
                },
                icon ?? /* @__PURE__ */ import_react13.default.createElement(import_ListOutlined3.default, null)
              ),
              /* @__PURE__ */ import_react13.default.createElement(
                import_ListItemText3.default,
                {
                  primary: label,
                  primaryTypographyProps: {
                    noWrap: true,
                    fontSize: "14px"
                  }
                }
              ),
              isOpen ? /* @__PURE__ */ import_react13.default.createElement(
                import_ExpandLess3.default,
                {
                  sx: {
                    color: "text.icon"
                  }
                }
              ) : /* @__PURE__ */ import_react13.default.createElement(
                import_ExpandMore3.default,
                {
                  sx: {
                    color: "text.icon"
                  }
                }
              )
            )
          ), !siderCollapsed && /* @__PURE__ */ import_react13.default.createElement(
            import_Collapse3.default,
            {
              in: open[item.key || ""],
              timeout: "auto",
              unmountOnExit: true
            },
            /* @__PURE__ */ import_react13.default.createElement(import_List3.default, { component: "div", disablePadding: true }, renderTreeView(children, selectedKey2))
          ))
        );
      }
      const linkStyle = activeItemDisabled && isSelected ? { pointerEvents: "none" } : {};
      return /* @__PURE__ */ import_react13.default.createElement(
        import_core9.CanAccess,
        {
          key: item.key,
          resource: name,
          action: "list",
          params: { resource: item }
        },
        /* @__PURE__ */ import_react13.default.createElement(
          import_Tooltip3.default,
          {
            title: label ?? name,
            placement: "right",
            disableHoverListener: !siderCollapsed,
            arrow: true
          },
          /* @__PURE__ */ import_react13.default.createElement(
            import_ListItemButton3.default,
            {
              component: ActiveLink,
              to: route,
              selected: isSelected,
              style: linkStyle,
              onClick: () => {
                setMobileSiderOpen(false);
              },
              sx: {
                pl: isNested ? 4 : 2,
                py: isNested ? 1.25 : 1,
                justifyContent: "center",
                color: isSelected ? "primary.main" : "text.primary"
              }
            },
            /* @__PURE__ */ import_react13.default.createElement(
              import_ListItemIcon3.default,
              {
                sx: {
                  justifyContent: "center",
                  transition: "margin-right 0.3s",
                  marginRight: siderCollapsed ? "0px" : "12px",
                  minWidth: "24px",
                  color: "currentColor"
                }
              },
              icon ?? /* @__PURE__ */ import_react13.default.createElement(import_ListOutlined3.default, null)
            ),
            /* @__PURE__ */ import_react13.default.createElement(
              import_ListItemText3.default,
              {
                primary: label,
                primaryTypographyProps: {
                  noWrap: true,
                  fontSize: "14px"
                }
              }
            )
          )
        )
      );
    });
  }, "renderTreeView");
  const dashboard = hasDashboard ? /* @__PURE__ */ import_react13.default.createElement(import_core9.CanAccess, { resource: "dashboard", action: "list" }, /* @__PURE__ */ import_react13.default.createElement(
    import_Tooltip3.default,
    {
      title: translate("dashboard.title", "Dashboard"),
      placement: "right",
      disableHoverListener: !siderCollapsed,
      arrow: true
    },
    /* @__PURE__ */ import_react13.default.createElement(
      import_ListItemButton3.default,
      {
        component: ActiveLink,
        to: "/",
        selected: selectedKey === "/",
        onClick: () => {
          setMobileSiderOpen(false);
        },
        sx: {
          pl: 2,
          py: 1,
          justifyContent: "center",
          color: selectedKey === "/" ? "primary.main" : "text.primary"
        }
      },
      /* @__PURE__ */ import_react13.default.createElement(
        import_ListItemIcon3.default,
        {
          sx: {
            justifyContent: "center",
            minWidth: "24px",
            transition: "margin-right 0.3s",
            marginRight: siderCollapsed ? "0px" : "12px",
            color: "currentColor",
            fontSize: "14px"
          }
        },
        /* @__PURE__ */ import_react13.default.createElement(import_Dashboard3.default, null)
      ),
      /* @__PURE__ */ import_react13.default.createElement(
        import_ListItemText3.default,
        {
          primary: translate("dashboard.title", "Dashboard"),
          primaryTypographyProps: {
            noWrap: true,
            fontSize: "14px"
          }
        }
      )
    )
  )) : null;
  const handleLogout = /* @__PURE__ */ __name(() => {
    if (warnWhen) {
      const confirm = window.confirm(
        t(
          "warnWhenUnsavedChanges",
          "Are you sure you want to leave? You have unsaved changes."
        )
      );
      if (confirm) {
        setWarnWhen(false);
        mutateLogout();
      }
    } else {
      mutateLogout();
    }
  }, "handleLogout");
  const logout = isExistAuthentication && /* @__PURE__ */ import_react13.default.createElement(
    import_Tooltip3.default,
    {
      title: t("buttons.logout", "Logout"),
      placement: "right",
      disableHoverListener: !siderCollapsed,
      arrow: true
    },
    /* @__PURE__ */ import_react13.default.createElement(
      import_ListItemButton3.default,
      {
        key: "logout",
        onClick: () => handleLogout(),
        sx: {
          justifyContent: "center"
        }
      },
      /* @__PURE__ */ import_react13.default.createElement(
        import_ListItemIcon3.default,
        {
          sx: {
            justifyContent: "center",
            minWidth: "24px",
            transition: "margin-right 0.3s",
            marginRight: siderCollapsed ? "0px" : "12px",
            color: "currentColor"
          }
        },
        /* @__PURE__ */ import_react13.default.createElement(import_Logout3.default, null)
      ),
      /* @__PURE__ */ import_react13.default.createElement(
        import_ListItemText3.default,
        {
          primary: t("buttons.logout", "Logout"),
          primaryTypographyProps: {
            noWrap: true,
            fontSize: "14px"
          }
        }
      )
    )
  );
  const items = renderTreeView(menuItems, selectedKey);
  const renderSider = /* @__PURE__ */ __name(() => {
    if (render) {
      return render({
        dashboard,
        logout,
        items,
        collapsed: siderCollapsed
      });
    }
    return /* @__PURE__ */ import_react13.default.createElement(import_react13.default.Fragment, null, dashboard, items, logout);
  }, "renderSider");
  const drawer = /* @__PURE__ */ import_react13.default.createElement(
    import_List3.default,
    {
      disablePadding: true,
      sx: {
        flexGrow: 1,
        paddingTop: "16px"
      }
    },
    renderSider()
  );
  return /* @__PURE__ */ import_react13.default.createElement(import_react13.default.Fragment, null, /* @__PURE__ */ import_react13.default.createElement(
    import_Box5.default,
    {
      sx: {
        width: { xs: drawerWidth() },
        display: {
          xs: "none",
          md: "block"
        },
        transition: "width 0.3s ease"
      }
    }
  ), /* @__PURE__ */ import_react13.default.createElement(
    import_Box5.default,
    {
      component: "nav",
      sx: {
        position: "fixed",
        zIndex: 1101,
        width: { sm: drawerWidth() },
        display: "flex"
      }
    },
    /* @__PURE__ */ import_react13.default.createElement(
      import_Drawer3.default,
      {
        variant: "temporary",
        elevation: 2,
        open: mobileSiderOpen,
        onClose: () => setMobileSiderOpen(false),
        ModalProps: {
          keepMounted: true
          // Better open performance on mobile.
        },
        sx: {
          display: {
            sm: "block",
            md: "none"
          }
        }
      },
      /* @__PURE__ */ import_react13.default.createElement(
        import_Box5.default,
        {
          sx: {
            width: drawerWidth()
          }
        },
        /* @__PURE__ */ import_react13.default.createElement(
          import_Box5.default,
          {
            sx: {
              height: 64,
              display: "flex",
              alignItems: "center",
              paddingLeft: "16px",
              fontSize: "14px"
            }
          },
          /* @__PURE__ */ import_react13.default.createElement(RenderToTitle, { collapsed: false })
        ),
        drawer
      )
    ),
    /* @__PURE__ */ import_react13.default.createElement(
      import_Drawer3.default,
      {
        variant: "permanent",
        sx: {
          display: { xs: "none", md: "block" },
          "& .MuiDrawer-paper": {
            width: drawerWidth(),
            overflow: "hidden",
            transition: "width 200ms cubic-bezier(0.4, 0, 0.6, 1) 0ms"
          }
        },
        open: true
      },
      /* @__PURE__ */ import_react13.default.createElement(
        import_Paper2.default,
        {
          elevation: 0,
          sx: {
            fontSize: "14px",
            width: "100%",
            height: 64,
            display: "flex",
            flexShrink: 0,
            alignItems: "center",
            justifyContent: siderCollapsed ? "center" : "space-between",
            paddingLeft: siderCollapsed ? 0 : "16px",
            paddingRight: siderCollapsed ? 0 : "8px",
            variant: "outlined",
            borderRadius: 0,
            borderBottom: (theme) => `1px solid ${theme.palette.action.focus}`
          }
        },
        /* @__PURE__ */ import_react13.default.createElement(RenderToTitle, { collapsed: siderCollapsed }),
        !siderCollapsed && /* @__PURE__ */ import_react13.default.createElement(import_IconButton4.default, { size: "small", onClick: () => setSiderCollapsed(true) }, /* @__PURE__ */ import_react13.default.createElement(import_ChevronLeft3.default, null))
      ),
      /* @__PURE__ */ import_react13.default.createElement(
        import_Box5.default,
        {
          sx: {
            flexGrow: 1,
            overflowX: "hidden",
            overflowY: "auto"
          }
        },
        drawer
      )
    )
  ));
}, "ThemedSiderV2");

// src/components/themedLayoutV2/header/index.tsx
var import_react15 = __toESM(require("react"));
var import_core10 = require("@refinedev/core");
var import_AppBar3 = __toESM(require("@mui/material/AppBar"));
var import_Stack3 = __toESM(require("@mui/material/Stack"));
var import_Toolbar3 = __toESM(require("@mui/material/Toolbar"));
var import_Typography4 = __toESM(require("@mui/material/Typography"));
var import_Avatar3 = __toESM(require("@mui/material/Avatar"));

// src/components/themedLayoutV2/hamburgerMenu/index.tsx
var import_react14 = __toESM(require("react"));
var import_IconButton5 = __toESM(require("@mui/material/IconButton"));
var import_Menu2 = __toESM(require("@mui/icons-material/Menu"));
var HamburgerIcon = /* @__PURE__ */ __name((props) => /* @__PURE__ */ import_react14.default.createElement(import_IconButton5.default, { color: "inherit", "aria-label": "open drawer", edge: "start", ...props }, /* @__PURE__ */ import_react14.default.createElement(import_Menu2.default, null)), "HamburgerIcon");
var HamburgerMenu = /* @__PURE__ */ __name(() => {
  const {
    siderCollapsed,
    setSiderCollapsed,
    mobileSiderOpen,
    setMobileSiderOpen
  } = useThemedLayoutContext();
  return /* @__PURE__ */ import_react14.default.createElement(import_react14.default.Fragment, null, /* @__PURE__ */ import_react14.default.createElement(
    HamburgerIcon,
    {
      onClick: () => setSiderCollapsed(!siderCollapsed),
      sx: {
        mr: 2,
        display: { xs: "none", md: "flex" },
        ...!siderCollapsed && { display: "none" }
      }
    }
  ), /* @__PURE__ */ import_react14.default.createElement(
    HamburgerIcon,
    {
      onClick: () => setMobileSiderOpen(!mobileSiderOpen),
      sx: {
        mr: 2,
        display: { xs: "flex", md: "none" },
        ...mobileSiderOpen && { display: "none" }
      }
    }
  ));
}, "HamburgerMenu");

// src/components/themedLayoutV2/header/index.tsx
var ThemedHeaderV2 = /* @__PURE__ */ __name(({
  isSticky,
  sticky
}) => {
  const authProvider = (0, import_core10.useActiveAuthProvider)();
  const { data: user } = (0, import_core10.useGetIdentity)({
    v3LegacyAuthProviderCompatible: Boolean(authProvider == null ? void 0 : authProvider.isLegacy)
  });
  const prefferedSticky = (0, import_core10.pickNotDeprecated)(sticky, isSticky) ?? true;
  return /* @__PURE__ */ import_react15.default.createElement(import_AppBar3.default, { position: prefferedSticky ? "sticky" : "relative" }, /* @__PURE__ */ import_react15.default.createElement(import_Toolbar3.default, null, /* @__PURE__ */ import_react15.default.createElement(HamburgerMenu, null), /* @__PURE__ */ import_react15.default.createElement(
    import_Stack3.default,
    {
      direction: "row",
      width: "100%",
      justifyContent: "flex-end",
      alignItems: "center"
    },
    /* @__PURE__ */ import_react15.default.createElement(
      import_Stack3.default,
      {
        direction: "row",
        gap: "16px",
        alignItems: "center",
        justifyContent: "center"
      },
      (user == null ? void 0 : user.name) && /* @__PURE__ */ import_react15.default.createElement(import_Typography4.default, { variant: "subtitle2", }, user == null ? void 0 : user.name),
      (user == null ? void 0 : user.avatar) && /* @__PURE__ */ import_react15.default.createElement(import_Avatar3.default, { src: user == null ? void 0 : user.avatar, alt: user == null ? void 0 : user.name })
    )
  )));
}, "ThemedHeaderV2");

// src/components/themedLayoutV2/index.tsx
var ThemedLayoutV2 = /* @__PURE__ */ __name(({
  Sider: Sider2,
  Header: Header2,
  Title: Title2,
  Footer,
  OffLayoutArea,
  children,
  initialSiderCollapsed
}) => {
  const SiderToRender = Sider2 ?? ThemedSiderV2;
  const HeaderToRender = Header2 ?? ThemedHeaderV2;
  return /* @__PURE__ */ import_react16.default.createElement(ThemedLayoutContextProvider, { initialSiderCollapsed }, /* @__PURE__ */ import_react16.default.createElement(import_Box6.default, { display: "flex", flexDirection: "row" }, /* @__PURE__ */ import_react16.default.createElement(SiderToRender, { Title: Title2 }), /* @__PURE__ */ import_react16.default.createElement(
    import_Box6.default,
    {
      sx: [
        {
          display: "flex",
          flexDirection: "column",
          flex: 1,
          minWidth: "1px",
          minHeight: "1px"
        }
      ]
    },
    /* @__PURE__ */ import_react16.default.createElement(HeaderToRender, null),
    /* @__PURE__ */ import_react16.default.createElement(
      import_Box6.default,
      {
        component: "main",
        sx: {
          p: { xs: 1, md: 2, lg: 3 },
          flexGrow: 1,
          bgcolor: (theme) => theme.palette.background.default
        }
      },
      children
    ),
    Footer && /* @__PURE__ */ import_react16.default.createElement(Footer, null)
  ), OffLayoutArea && /* @__PURE__ */ import_react16.default.createElement(OffLayoutArea, null)));
}, "ThemedLayoutV2");

// src/components/themedLayoutV2/title/index.tsx
var import_react17 = __toESM(require("react"));
var import_core11 = require("@refinedev/core");
var import_Link2 = __toESM(require("@mui/material/Link"));
var import_SvgIcon2 = __toESM(require("@mui/material/SvgIcon"));
var import_Typography5 = __toESM(require("@mui/material/Typography"));
var ThemedTitleV2 = /* @__PURE__ */ __name(({
  collapsed,
  wrapperStyles,
  icon: iconFromProps,
  text: textFromProps
}) => {
  const {
    title: { icon: defaultIcon2, text: defaultText2 } = {}
  } = (0, import_core11.useRefineOptions)();
  const icon = typeof iconFromProps === "undefined" ? defaultIcon2 : iconFromProps;
  const text = typeof textFromProps === "undefined" ? defaultText2 : textFromProps;
  const routerType = (0, import_core11.useRouterType)();
  const Link6 = (0, import_core11.useLink)();
  const { Link: LegacyLink } = (0, import_core11.useRouterContext)();
  const ActiveLink = routerType === "legacy" ? LegacyLink : Link6;
  return /* @__PURE__ */ import_react17.default.createElement(
    import_Link2.default,
    {
      to: "/",
      component: ActiveLink,
      underline: "none",
      sx: {
        display: "flex",
        alignItems: "center",
        gap: "12px",
        ...wrapperStyles
      }
    },
    /* @__PURE__ */ import_react17.default.createElement(import_SvgIcon2.default, { height: "24px", width: "24px", color: "primary" }, icon),
    !collapsed && /* @__PURE__ */ import_react17.default.createElement(
      import_Typography5.default,
      {
        variant: "h6",
        fontWeight: 700,
        color: "text.primary",
        fontSize: "inherit",
        textOverflow: "ellipsis",
        overflow: "hidden"
      },
      text
    )
  );
}, "ThemedTitleV2");

// src/components/pages/login/index.tsx
var React15 = __toESM(require("react"));
var import_core12 = require("@refinedev/core");
var import_react_hook_form = require("@refinedev/react-hook-form");
var import_Button3 = __toESM(require("@mui/material/Button"));
var import_TextField = __toESM(require("@mui/material/TextField"));
var import_FormControlLabel = __toESM(require("@mui/material/FormControlLabel"));
var import_Checkbox = __toESM(require("@mui/material/Checkbox"));
var import_Link3 = __toESM(require("@mui/material/Link"));
var import_Box7 = __toESM(require("@mui/material/Box"));
var import_Typography6 = __toESM(require("@mui/material/Typography"));
var import_Container = __toESM(require("@mui/material/Container"));
var import_Card = __toESM(require("@mui/material/Card"));
var import_CardContent = __toESM(require("@mui/material/CardContent"));
var import_core13 = require("@refinedev/core");
var LoginPage = /* @__PURE__ */ __name(() => {
  var _a;
  const {
    register,
    handleSubmit,
    formState: { errors }
  } = (0, import_react_hook_form.useForm)();
  const authProvider = (0, import_core12.useActiveAuthProvider)();
  const { mutate: login, isLoading } = (0, import_core13.useLogin)({
    v3LegacyAuthProviderCompatible: Boolean(authProvider == null ? void 0 : authProvider.isLegacy)
  });
  const translate = (0, import_core13.useTranslate)();
  return /* @__PURE__ */ React15.createElement(React15.Fragment, null, /* @__PURE__ */ React15.createElement(
    import_Box7.default,
    {
      component: "div",
      sx: {
        background: "radial-gradient(50% 50% at 50% 50%, #63386A 0%, #310438 100%)",
        backgroundSize: "cover"
      }
    },
    /* @__PURE__ */ React15.createElement(
      import_Container.default,
      {
        component: "main",
        maxWidth: "xs",
        sx: {
          display: "flex",
          flexDirection: "column",
          justifyContent: "center",
          height: "100vh"
        }
      },
      /* @__PURE__ */ React15.createElement(
        import_Box7.default,
        {
          sx: {
            display: "flex",
            justifyContent: "center",
            flexDirection: "column",
            alignItems: "center"
          }
        },
        /* @__PURE__ */ React15.createElement("div", null, /* @__PURE__ */ React15.createElement(
          "img",
          {
            src: "https://refine.ams3.cdn.digitaloceanspaces.com/logo/refine.svg",
            alt: "Refine Logo",
            style: { maxWidth: 200 }
          }
        )),
        /* @__PURE__ */ React15.createElement(import_Box7.default, { mt: 4 }, /* @__PURE__ */ React15.createElement(import_Card.default, null, /* @__PURE__ */ React15.createElement(import_CardContent.default, { sx: { paddingX: "32px" } }, /* @__PURE__ */ React15.createElement(
          import_Typography6.default,
          {
            component: "h1",
            variant: "h5",
            align: "center",
            sx: {
              fontWeight: "700",
              margin: "12px 0"
            }
          },
          translate("pages.login.title", "Sign in your account")
        ), /* @__PURE__ */ React15.createElement(
          import_Box7.default,
          {
            component: "form",
            onSubmit: handleSubmit((data) => {
              login(data);
            }),
            gap: "16px"
          },
          /* @__PURE__ */ React15.createElement(
            import_TextField.default,
            {
              ...register("username", {
                required: true
              }),
              id: "username",
              margin: "normal",
              size: "small",
              fullWidth: true,
              label: translate("pages.login.username", "Username"),
              name: "username",
              autoComplete: "username",
              autoFocus: true
            }
          ),
          /* @__PURE__ */ React15.createElement(
            import_TextField.default,
            {
              ...register("password", {
                required: true
              }),
              id: "password",
              size: "small",
              margin: "normal",
              fullWidth: true,
              name: "password",
              label: translate("pages.login.password", "Password"),
              helperText: (_a = errors == null ? void 0 : errors.password) == null ? void 0 : _a.message,
              type: "password",
              placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF",
              autoComplete: "current-password"
            }
          ),
          /* @__PURE__ */ React15.createElement(
            import_Box7.default,
            {
              component: "div",
              sx: {
                display: "flex",
                justifyContent: "space-between",
                alignItems: "center"
              }
            },
            /* @__PURE__ */ React15.createElement(
              import_FormControlLabel.default,
              {
                sx: {
                  span: {
                    fontSize: "12px",
                    color: "text.secondary"
                  }
                },
                color: "secondary",
                control: /* @__PURE__ */ React15.createElement(
                  import_Checkbox.default,
                  {
                    size: "small",
                    id: "remember",
                    ...register("remember")
                  }
                ),
                label: translate("pages.login.remember", "Remember me")
              }
            ),
            /* @__PURE__ */ React15.createElement(
              import_Link3.default,
              {
                href: "#",
                sx: {
                  fontSize: "12px",
                  textDecoration: "none"
                }
              },
              /* @__PURE__ */ React15.createElement(
                import_Typography6.default,
                {
                  sx: {
                    fontSize: "12px"
                  }
                },
                translate(
                  "pages.login.forgotPassword",
                  "Forgot password?"
                )
              )
            )
          ),
          /* @__PURE__ */ React15.createElement(
            import_Button3.default,
            {
              type: "submit",
              fullWidth: true,
              variant: "contained",
              sx: {
                my: "8px",
                color: "white"
              },
              disabled: isLoading
            },
            translate("pages.login.signin", "Sign in")
          ),
          /* @__PURE__ */ React15.createElement(import_Box7.default, { style: { marginTop: 8 } }, /* @__PURE__ */ React15.createElement(import_Typography6.default, { variant: "subtitle2" }, translate(
            "pages.login.noAccount",
            "Don\u2019t have an account?"
          ), " ", /* @__PURE__ */ React15.createElement(
            import_Link3.default,
            {
              underline: "none",
              href: "#",
              style: {
                fontWeight: "bold"
              }
            },
            translate("pages.login.signup", "Sign up")
          )))
        ))))
      )
    )
  ));
}, "LoginPage");

// src/components/pages/ready/index.tsx
var React16 = __toESM(require("react"));
var import_Stack4 = __toESM(require("@mui/material/Stack"));
var import_Grid = __toESM(require("@mui/material/Grid"));
var import_Typography7 = __toESM(require("@mui/material/Typography"));
var import_Button4 = __toESM(require("@mui/material/Button"));
var import_Link4 = __toESM(require("@mui/material/Link"));
var ReadyPage = /* @__PURE__ */ __name(() => {
  const renderCode = /* @__PURE__ */ __name((text) => /* @__PURE__ */ React16.createElement(
    import_Typography7.default,
    {
      sx: {
        backgroundColor: (theme) => theme.palette.secondary.contrastText,
        color: (theme) => theme.palette.secondary.main,
        display: "inline-block",
        fontFamily: "monospace"
      }
    },
    text
  ), "renderCode");
  return /* @__PURE__ */ React16.createElement(React16.Fragment, null, /* @__PURE__ */ React16.createElement(
    import_Grid.default,
    {
      container: true,
      sx: {
        backgroundColor: (theme) => theme.palette.secondary.main,
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
        height: "100vh",
        backgroundImage: "url('https://refine.ams3.cdn.digitaloceanspaces.com/login-background/background.png')",
        backgroundSize: "cover"
      },
      p: 3
    },
    /* @__PURE__ */ React16.createElement(
      import_Grid.default,
      {
        display: "flex",
        flexDirection: "column",
        flex: 1,
        alignItems: "center",
        color: (theme) => theme.palette.primary.contrastText
      },
      /* @__PURE__ */ React16.createElement(import_Typography7.default, { display: "flex", justifyContent: "center" }, /* @__PURE__ */ React16.createElement(
        "img",
        {
          style: { marginBottom: "48px" },
          src: "https://refine.ams3.cdn.digitaloceanspaces.com/logo/refine.svg",
          alt: "Refine Logo"
        }
      )),
      /* @__PURE__ */ React16.createElement(
        import_Typography7.default,
        {
          variant: "h2",
          fontWeight: "bold",
          display: "flex",
          justifyContent: "center"
        },
        "Welcome on board"
      ),
      /* @__PURE__ */ React16.createElement(import_Typography7.default, { variant: "h5", mt: 2 }, "Your configuration is completed."),
      /* @__PURE__ */ React16.createElement(import_Typography7.default, { variant: "h6", mt: 2 }, "Now you can get started by adding your resources to the", " ", renderCode("resources"), " property of ", renderCode("Refine")),
      /* @__PURE__ */ React16.createElement(import_Grid.default, { item: true, sm: 12, mt: 12, display: "flex", justifyContent: "center" }, /* @__PURE__ */ React16.createElement(import_Stack4.default, { spacing: 2, direction: { xs: "column", sm: "row" } }, /* @__PURE__ */ React16.createElement(
        import_Link4.default,
        {
          href: "https://refine.dev",
          target: "_blank",
          rel: "noreferrer",
          style: {
            textDecoration: "none"
          }
        },
        /* @__PURE__ */ React16.createElement(import_Button4.default, { variant: "outlined", fullWidth: true }, "Documentation")
      ), /* @__PURE__ */ React16.createElement(
        import_Link4.default,
        {
          href: "https://refine.dev/examples",
          target: "_blank",
          rel: "noreferrer",
          style: {
            textDecoration: "none"
          }
        },
        /* @__PURE__ */ React16.createElement(import_Button4.default, { variant: "outlined", fullWidth: true }, "Examples")
      ), /* @__PURE__ */ React16.createElement(
        import_Link4.default,
        {
          href: "https://discord.gg/refine",
          target: "_blank",
          rel: "noreferrer",
          style: {
            textDecoration: "none"
          }
        },
        /* @__PURE__ */ React16.createElement(import_Button4.default, { variant: "outlined", fullWidth: true }, "Community")
      )))
    )
  ));
}, "ReadyPage");

// src/components/pages/welcome/index.tsx
var import_react18 = __toESM(require("react"));
var import_core14 = require("@refinedev/core");
var WelcomePage = /* @__PURE__ */ __name(() => {
  return /* @__PURE__ */ import_react18.default.createElement(import_core14.WelcomePage, null);
}, "WelcomePage");

// src/components/pages/error/index.tsx
var import_react19 = __toESM(require("react"));
var import_core15 = require("@refinedev/core");
var import_core16 = require("@refinedev/core");
var import_Stack5 = __toESM(require("@mui/material/Stack"));
var import_Button5 = __toESM(require("@mui/material/Button"));
var import_Tooltip4 = __toESM(require("@mui/material/Tooltip"));
var import_Typography8 = __toESM(require("@mui/material/Typography"));
var import_Grid2 = __toESM(require("@mui/material/Grid"));
var import_Info = __toESM(require("@mui/icons-material/Info"));
var ErrorComponent = /* @__PURE__ */ __name(() => {
  const [errorMessage, setErrorMessage] = (0, import_react19.useState)();
  const { push } = (0, import_core16.useNavigation)();
  const go = (0, import_core15.useGo)();
  const routerType = (0, import_core15.useRouterType)();
  const { resource, action } = (0, import_core15.useResource)();
  const translate = (0, import_core16.useTranslate)();
  (0, import_react19.useEffect)(() => {
    if (resource && action) {
      setErrorMessage(
        translate(
          "pages.error.info",
          {
            action,
            resource: resource == null ? void 0 : resource.name
          },
          `You may have forgotten to add the "${action}" component to "${resource == null ? void 0 : resource.name}" resource.`
        )
      );
    }
  }, [action, resource]);
  return /* @__PURE__ */ import_react19.default.createElement(import_Grid2.default, { display: "flex", justifyContent: "center", alignItems: "center", mt: 20 }, /* @__PURE__ */ import_react19.default.createElement(import_Grid2.default, { container: true, direction: "column", display: "flex", alignItems: "center" }, /* @__PURE__ */ import_react19.default.createElement(import_Typography8.default, { variant: "h1" }, "404"), /* @__PURE__ */ import_react19.default.createElement(import_Stack5.default, { direction: "row", spacing: "2" }, /* @__PURE__ */ import_react19.default.createElement(import_Typography8.default, null, translate(
    "pages.error.404",
    "Sorry, the page you visited does not exist."
  )), errorMessage && /* @__PURE__ */ import_react19.default.createElement(import_Tooltip4.default, { title: errorMessage }, /* @__PURE__ */ import_react19.default.createElement(import_Info.default, { }))), /* @__PURE__ */ import_react19.default.createElement(
    import_Button5.default,
    {
      onClick: () => {
        if (routerType === "legacy") {
          push("/");
        } else {
          go({ to: "/" });
        }
      }
    },
    translate("pages.error.backHome", "Back Home")
  )));
}, "ErrorComponent");

// src/components/pages/auth/index.tsx
var import_react20 = __toESM(require("react"));

// src/components/pages/auth/components/login/index.tsx
var React19 = __toESM(require("react"));
var import_core17 = require("@refinedev/core");
var import_react_hook_form2 = require("@refinedev/react-hook-form");
var import_react_hook_form3 = require("react-hook-form");
var import_Button6 = __toESM(require("@mui/material/Button"));
var import_TextField2 = __toESM(require("@mui/material/TextField"));
var import_Box8 = __toESM(require("@mui/material/Box"));
var import_Typography9 = __toESM(require("@mui/material/Typography"));
var import_Container2 = __toESM(require("@mui/material/Container"));
var import_Card2 = __toESM(require("@mui/material/Card"));
var import_CardContent2 = __toESM(require("@mui/material/CardContent"));
var import_Checkbox2 = __toESM(require("@mui/material/Checkbox"));
var import_FormControlLabel2 = __toESM(require("@mui/material/FormControlLabel"));
var import_Divider = __toESM(require("@mui/material/Divider"));
var import_Link5 = __toESM(require("@mui/material/Link"));
var import_Stack6 = __toESM(require("@mui/material/Stack"));
var import_core18 = require("@refinedev/core");

// src/components/pages/auth/components/styles.ts
var layoutStyles = {};
var titleStyles = {
  textAlign: "center",
  fontSize: "24px",
  marginBottom: "24px",
  overflowWrap: "break-word",
  hyphens: "manual",
  textOverflow: "unset",
  whiteSpace: "pre-wrap"
};

// src/components/pages/auth/components/login/index.tsx
var LoginPage2 = /* @__PURE__ */ __name(({
  providers,
  registerLink,
  forgotPasswordLink,
  rememberMe,
  contentProps,
  wrapperProps,
  renderContent,
  formProps,
  title,
  hideForm
}) => {
  var _a;
  const { onSubmit, ...useFormProps } = formProps || {};
  const methods = (0, import_react_hook_form2.useForm)({
    ...useFormProps
  });
  const {
    register,
    handleSubmit,
    formState: { errors }
  } = methods;
  const authProvider = (0, import_core17.useActiveAuthProvider)();
  const { mutate: login, isLoading } = (0, import_core18.useLogin)({
    v3LegacyAuthProviderCompatible: Boolean(authProvider == null ? void 0 : authProvider.isLegacy)
  });
  const translate = (0, import_core18.useTranslate)();
  const routerType = (0, import_core18.useRouterType)();
  const Link6 = (0, import_core18.useLink)();
  const { Link: LegacyLink } = (0, import_core18.useRouterContext)();
  const ActiveLink = routerType === "legacy" ? LegacyLink : Link6;
  const PageTitle = title === false ? null : /* @__PURE__ */ React19.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ React19.createElement(
      ThemedTitleV2,
      {
        collapsed: false,
        wrapperStyles: {
          gap: "8px"
        }
      }
    )
  );
  const renderProviders = /* @__PURE__ */ __name(() => {
    if (providers && providers.length > 0) {
      return /* @__PURE__ */ React19.createElement(React19.Fragment, null, /* @__PURE__ */ React19.createElement(import_Stack6.default, { spacing: 1 }, providers.map((provider) => {
        return /* @__PURE__ */ React19.createElement(
          import_Button6.default,
          {
            key: provider.name,
            variant: "outlined",
            fullWidth: true,
            sx: {
              color: "primary.light",
              borderColor: "primary.light",
              textTransform: "none"
            },
            onClick: () => login({ providerName: provider.name }),
            startIcon: provider.icon
          },
          provider.label
        );
      })), !hideForm && /* @__PURE__ */ React19.createElement(
        import_Divider.default,
        {
          sx: {
            fontSize: "12px",
            marginY: "16px"
          }
        },
        translate("pages.login.divider", "or")
      ));
    }
    return null;
  }, "renderProviders");
  const Content = /* @__PURE__ */ React19.createElement(import_Card2.default, { ...contentProps ?? {} }, /* @__PURE__ */ React19.createElement(import_CardContent2.default, { sx: { p: "32px", "&:last-child": { pb: "32px" } } }, /* @__PURE__ */ React19.createElement(
    import_Typography9.default,
    {
      component: "h1",
      variant: "h5",
      align: "center",
      style: titleStyles,
      color: "primary",
      fontWeight: 700
    },
    translate("pages.login.title", "Sign in to your account")
  ), renderProviders(), !hideForm && /* @__PURE__ */ React19.createElement(
    import_Box8.default,
    {
      component: "form",
      onSubmit: handleSubmit((data) => {
        if (onSubmit) {
          return onSubmit(data);
        }
        return login(data);
      })
    },
    /* @__PURE__ */ React19.createElement(
      import_TextField2.default,
      {
        ...register("email", {
          required: translate(
            "pages.login.errors.requiredEmail",
            "Email is required"
          )
        }),
        id: "email",
        margin: "normal",
        fullWidth: true,
        label: translate("pages.login.fields.email", "Email"),
        error: !!errors.email,
        name: "email",
        type: "email",
        autoComplete: "email",
        sx: {
          mt: 0
        }
      }
    ),
    /* @__PURE__ */ React19.createElement(
      import_TextField2.default,
      {
        ...register("password", {
          required: translate(
            "pages.login.errors.requiredPassword",
            "Password is required"
          )
        }),
        id: "password",
        margin: "normal",
        fullWidth: true,
        name: "password",
        label: translate("pages.login.fields.password", "Password"),
        helperText: (_a = errors == null ? void 0 : errors.password) == null ? void 0 : _a.message,
        error: !!errors.password,
        type: "password",
        placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF",
        autoComplete: "current-password",
        sx: {
          mb: 0
        }
      }
    ),
    /* @__PURE__ */ React19.createElement(
      import_Box8.default,
      {
        component: "div",
        sx: {
          mt: "24px",
          display: "flex",
          justifyContent: "space-between",
          alignItems: "center"
        }
      },
      rememberMe ?? /* @__PURE__ */ React19.createElement(
        import_FormControlLabel2.default,
        {
          sx: {
            span: {
              fontSize: "14px",
              color: "text.secondary"
            }
          },
          color: "secondary",
          control: /* @__PURE__ */ React19.createElement(
            import_Checkbox2.default,
            {
              size: "small",
              id: "remember",
              ...register("remember")
            }
          ),
          label: translate(
            "pages.login.buttons.rememberMe",
            "Remember me"
          )
        }
      ),
      forgotPasswordLink ?? /* @__PURE__ */ React19.createElement(
        import_Link5.default,
        {
          variant: "body2",
          color: "primary",
          fontSize: "12px",
          component: ActiveLink,
          underline: "none",
          to: "/forgot-password"
        },
        translate(
          "pages.login.buttons.forgotPassword",
          "Forgot password?"
        )
      )
    ),
    /* @__PURE__ */ React19.createElement(
      import_Button6.default,
      {
        type: "submit",
        fullWidth: true,
        variant: "contained",
        disabled: isLoading,
        sx: { mt: "24px" }
      },
      translate("pages.login.signin", "Sign in")
    )
  ), registerLink ?? /* @__PURE__ */ React19.createElement(
    import_Box8.default,
    {
      sx: {
        mt: "24px",
        display: "flex",
        justifyContent: "center",
        alignItems: "center"
      }
    },
    /* @__PURE__ */ React19.createElement(
      import_Typography9.default,
      {
        textAlign: "center",
        variant: "body2",
        component: "span",
        fontSize: "12px"
      },
      translate(
        "pages.login.buttons.noAccount",
        "Don\u2019t have an account?"
      )
    ),
    /* @__PURE__ */ React19.createElement(
      import_Link5.default,
      {
        ml: "4px",
        fontSize: "12px",
        variant: "body2",
        color: "primary",
        component: ActiveLink,
        underline: "none",
        to: "/register",
        fontWeight: "bold"
      },
      translate("pages.login.signup", "Sign up")
    )
  )));
  return /* @__PURE__ */ React19.createElement(import_react_hook_form3.FormProvider, { ...methods }, /* @__PURE__ */ React19.createElement(import_Box8.default, { component: "div", style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ React19.createElement(
    import_Container2.default,
    {
      component: "main",
      sx: {
        display: "flex",
        flexDirection: "column",
        justifyContent: hideForm ? "flex-start" : "center",
        alignItems: "center",
        minHeight: "100dvh",
        padding: "16px",
        width: "100%",
        maxWidth: "400px"
      }
    },
    /* @__PURE__ */ React19.createElement(
      import_Box8.default,
      {
        sx: {
          width: "100%",
          maxWidth: "400px",
          display: "flex",
          flexDirection: "column",
          paddingTop: hideForm ? "15dvh" : 0
        }
      },
      renderContent ? renderContent(Content, PageTitle) : /* @__PURE__ */ React19.createElement(React19.Fragment, null, PageTitle, Content)
    )
  )));
}, "LoginPage");

// src/components/pages/auth/components/register/index.tsx
var React20 = __toESM(require("react"));
var import_core19 = require("@refinedev/core");
var import_react_hook_form4 = require("@refinedev/react-hook-form");
var import_Button7 = __toESM(require("@mui/material/Button"));
var import_TextField3 = __toESM(require("@mui/material/TextField"));
var import_Box9 = __toESM(require("@mui/material/Box"));
var import_Typography10 = __toESM(require("@mui/material/Typography"));
var import_Container3 = __toESM(require("@mui/material/Container"));
var import_Card3 = __toESM(require("@mui/material/Card"));
var import_CardContent3 = __toESM(require("@mui/material/CardContent"));
var import_Divider2 = __toESM(require("@mui/material/Divider"));
var import_Stack7 = __toESM(require("@mui/material/Stack"));
var import_Link6 = __toESM(require("@mui/material/Link"));
var import_core20 = require("@refinedev/core");
var RegisterPage = /* @__PURE__ */ __name(({
  loginLink,
  wrapperProps,
  contentProps,
  renderContent,
  providers,
  formProps,
  title,
  hideForm
}) => {
  const { onSubmit, ...useFormProps } = formProps || {};
  const {
    register,
    handleSubmit,
    formState: { errors }
  } = (0, import_react_hook_form4.useForm)({
    ...useFormProps
  });
  const authProvider = (0, import_core19.useActiveAuthProvider)();
  const { mutate: registerMutate, isLoading } = (0, import_core20.useRegister)({
    v3LegacyAuthProviderCompatible: Boolean(authProvider == null ? void 0 : authProvider.isLegacy)
  });
  const translate = (0, import_core20.useTranslate)();
  const routerType = (0, import_core20.useRouterType)();
  const Link6 = (0, import_core20.useLink)();
  const { Link: LegacyLink } = (0, import_core20.useRouterContext)();
  const ActiveLink = routerType === "legacy" ? LegacyLink : Link6;
  const PageTitle = title === false ? null : /* @__PURE__ */ React20.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ React20.createElement(
      ThemedTitleV2,
      {
        collapsed: false,
        wrapperStyles: {
          gap: "8px"
        }
      }
    )
  );
  const renderProviders = /* @__PURE__ */ __name(() => {
    if (providers && providers.length > 0) {
      return /* @__PURE__ */ React20.createElement(React20.Fragment, null, /* @__PURE__ */ React20.createElement(import_Stack7.default, { spacing: 1 }, providers.map((provider) => {
        return /* @__PURE__ */ React20.createElement(
          import_Button7.default,
          {
            key: provider.name,
            color: "secondary",
            fullWidth: true,
            variant: "outlined",
            sx: {
              color: "primary.light",
              borderColor: "primary.light",
              textTransform: "none"
            },
            onClick: () => registerMutate({
              providerName: provider.name
            }),
            startIcon: provider.icon
          },
          provider.label
        );
      })), !hideForm && /* @__PURE__ */ React20.createElement(
        import_Divider2.default,
        {
          sx: {
            fontSize: "12px",
            marginY: "16px"
          }
        },
        translate(
          "pages.register.divider",
          translate("pages.login.divider", "or")
        )
      ));
    }
    return null;
  }, "renderProviders");
  const Content = /* @__PURE__ */ React20.createElement(import_Card3.default, { ...contentProps ?? {} }, /* @__PURE__ */ React20.createElement(import_CardContent3.default, { sx: { p: "32px", "&:last-child": { pb: "32px" } } }, /* @__PURE__ */ React20.createElement(
    import_Typography10.default,
    {
      component: "h1",
      variant: "h5",
      align: "center",
      style: titleStyles,
      color: "primary",
      fontWeight: 700
    },
    translate("pages.register.title", "Sign up for your account")
  ), renderProviders(), !hideForm && /* @__PURE__ */ React20.createElement(
    import_Box9.default,
    {
      component: "form",
      onSubmit: handleSubmit((data) => {
        if (onSubmit) {
          return onSubmit(data);
        }
        return registerMutate(data);
      })
    },
    /* @__PURE__ */ React20.createElement(
      import_TextField3.default,
      {
        ...register("email", {
          required: translate(
            "pages.register.errors.requiredEmail",
            "Email is required"
          ),
          pattern: {
            value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
            message: translate(
              "pages.register.errors.validEmail",
              "Invalid email address"
            )
          }
        }),
        id: "email",
        margin: "normal",
        fullWidth: true,
        label: translate("pages.register.email", "Email"),
        error: !!errors.email,
        helperText: errors["email"] ? errors["email"].message : "",
        name: "email",
        autoComplete: "email",
        sx: {
          mt: 0
        }
      }
    ),
    /* @__PURE__ */ React20.createElement(
      import_TextField3.default,
      {
        ...register("password", {
          required: translate(
            "pages.register.errors.requiredPassword",
            "Password is required"
          )
        }),
        id: "password",
        margin: "normal",
        fullWidth: true,
        name: "password",
        label: translate("pages.register.fields.password", "Password"),
        helperText: errors["password"] ? errors["password"].message : "",
        error: !!errors.password,
        type: "password",
        placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF",
        autoComplete: "current-password",
        sx: {
          mb: 0
        }
      }
    ),
    /* @__PURE__ */ React20.createElement(
      import_Button7.default,
      {
        type: "submit",
        fullWidth: true,
        variant: "contained",
        disabled: isLoading,
        sx: {
          mt: "24px"
        }
      },
      translate("pages.register.signup", "Sign up")
    )
  ), loginLink ?? /* @__PURE__ */ React20.createElement(
    import_Box9.default,
    {
      display: "flex",
      justifyContent: "flex-end",
      alignItems: "center",
      sx: {
        mt: "24px",
        display: "flex",
        justifyContent: "center",
        alignItems: "center"
      }
    },
    /* @__PURE__ */ React20.createElement(import_Typography10.default, { variant: "body2", component: "span", fontSize: "12px" }, translate(
      "pages.register.buttons.haveAccount",
      translate(
        "pages.login.buttons.haveAccount",
        "Have an account?"
      )
    )),
    /* @__PURE__ */ React20.createElement(
      import_Link6.default,
      {
        ml: "4px",
        variant: "body2",
        color: "primary",
        component: ActiveLink,
        underline: "none",
        to: "/login",
        fontSize: "12px",
        fontWeight: "bold"
      },
      translate(
        "pages.register.signin",
        translate("pages.login.signin", "Sign in")
      )
    )
  )));
  return /* @__PURE__ */ React20.createElement(import_Box9.default, { component: "div", style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ React20.createElement(
    import_Container3.default,
    {
      component: "main",
      sx: {
        display: "flex",
        flexDirection: "column",
        justifyContent: hideForm ? "flex-start" : "center",
        alignItems: "center",
        minHeight: "100dvh",
        padding: "16px",
        width: "100%",
        maxWidth: "400px"
      }
    },
    /* @__PURE__ */ React20.createElement(
      import_Box9.default,
      {
        sx: {
          width: "100%",
          maxWidth: "400px",
          display: "flex",
          flexDirection: "column",
          paddingTop: hideForm ? "15dvh" : 0
        }
      },
      renderContent ? renderContent(Content, PageTitle) : /* @__PURE__ */ React20.createElement(React20.Fragment, null, PageTitle, Content)
    )
  ));
}, "RegisterPage");

// src/components/pages/auth/components/forgotPassword/index.tsx
var import_Button8 = __toESM(require("@mui/material/Button"));
var import_TextField4 = __toESM(require("@mui/material/TextField"));
var import_Box10 = __toESM(require("@mui/material/Box"));
var import_Typography11 = __toESM(require("@mui/material/Typography"));
var import_Container4 = __toESM(require("@mui/material/Container"));
var import_Card4 = __toESM(require("@mui/material/Card"));
var import_CardContent4 = __toESM(require("@mui/material/CardContent"));
var import_Link7 = __toESM(require("@mui/material/Link"));
var import_react_hook_form5 = require("@refinedev/react-hook-form");
var React21 = __toESM(require("react"));
var import_core21 = require("@refinedev/core");
var ForgotPasswordPage = /* @__PURE__ */ __name(({
  loginLink,
  wrapperProps,
  contentProps,
  renderContent,
  formProps,
  title
}) => {
  const { onSubmit, ...useFormProps } = formProps || {};
  const {
    register,
    handleSubmit,
    formState: { errors }
  } = (0, import_react_hook_form5.useForm)({
    ...useFormProps
  });
  const { mutate, isLoading } = (0, import_core21.useForgotPassword)();
  const translate = (0, import_core21.useTranslate)();
  const routerType = (0, import_core21.useRouterType)();
  const Link6 = (0, import_core21.useLink)();
  const { Link: LegacyLink } = (0, import_core21.useRouterContext)();
  const ActiveLink = routerType === "legacy" ? LegacyLink : Link6;
  const PageTitle = title === false ? null : /* @__PURE__ */ React21.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ React21.createElement(
      ThemedTitleV2,
      {
        collapsed: false,
        wrapperStyles: {
          gap: "8px"
        }
      }
    )
  );
  const Content = /* @__PURE__ */ React21.createElement(import_Card4.default, { ...contentProps ?? {} }, /* @__PURE__ */ React21.createElement(import_CardContent4.default, { sx: { p: "32px", "&:last-child": { pb: "32px" } } }, /* @__PURE__ */ React21.createElement(
    import_Typography11.default,
    {
      component: "h1",
      variant: "h5",
      align: "center",
      style: titleStyles,
      color: "primary",
      fontWeight: 700
    },
    translate("pages.forgotPassword.title", "Forgot your password?")
  ), /* @__PURE__ */ React21.createElement(
    import_Box10.default,
    {
      component: "form",
      onSubmit: handleSubmit((data) => {
        if (onSubmit) {
          return onSubmit(data);
        }
        return mutate(data);
      })
    },
    /* @__PURE__ */ React21.createElement(
      import_TextField4.default,
      {
        ...register("email", {
          required: translate(
            "pages.forgotPassword.errors.requiredEmail",
            "Email is required"
          ),
          pattern: {
            value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
            message: translate(
              "pages.register.errors.validEmail",
              "Invalid email address"
            )
          }
        }),
        id: "email",
        margin: "normal",
        fullWidth: true,
        label: translate("pages.forgotPassword.fields.email", "Email"),
        name: "email",
        type: "email",
        error: !!errors.email,
        autoComplete: "email",
        sx: {
          m: 0
        }
      }
    ),
    loginLink ?? /* @__PURE__ */ React21.createElement(import_Box10.default, { textAlign: "right", sx: { mt: "24px" } }, /* @__PURE__ */ React21.createElement(import_Typography11.default, { variant: "body2", component: "span", fontSize: "12px" }, translate(
      "pages.forgotPassword.buttons.haveAccount",
      translate(
        "pages.register.buttons.haveAccount",
        "Have an account? "
      )
    )), " ", /* @__PURE__ */ React21.createElement(
      import_Link7.default,
      {
        variant: "body2",
        component: ActiveLink,
        underline: "none",
        to: "/login",
        fontWeight: "bold",
        fontSize: "12px",
        color: "primary.light"
      },
      translate(
        "pages.forgotPassword.signin",
        translate("pages.login.signin", "Sign in")
      )
    )),
    /* @__PURE__ */ React21.createElement(
      import_Button8.default,
      {
        type: "submit",
        fullWidth: true,
        variant: "contained",
        sx: { mt: "24px" },
        disabled: isLoading
      },
      translate(
        "pages.forgotPassword.buttons.submit",
        "Send reset instructions"
      )
    )
  )));
  return /* @__PURE__ */ React21.createElement(React21.Fragment, null, /* @__PURE__ */ React21.createElement(import_Box10.default, { component: "div", style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ React21.createElement(
    import_Container4.default,
    {
      component: "main",
      maxWidth: "xs",
      sx: {
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        minHeight: "100dvh",
        padding: "16px",
        width: "100%",
        maxWidth: "400px"
      }
    },
    renderContent ? renderContent(Content, PageTitle) : /* @__PURE__ */ React21.createElement(React21.Fragment, null, PageTitle, Content)
  )));
}, "ForgotPasswordPage");

// src/components/pages/auth/components/updatePassword/index.tsx
var React22 = __toESM(require("react"));
var import_core22 = require("@refinedev/core");
var import_react_hook_form6 = require("@refinedev/react-hook-form");
var import_Button9 = __toESM(require("@mui/material/Button"));
var import_TextField5 = __toESM(require("@mui/material/TextField"));
var import_Box11 = __toESM(require("@mui/material/Box"));
var import_Typography12 = __toESM(require("@mui/material/Typography"));
var import_Container5 = __toESM(require("@mui/material/Container"));
var import_Card5 = __toESM(require("@mui/material/Card"));
var import_CardContent5 = __toESM(require("@mui/material/CardContent"));
var import_core23 = require("@refinedev/core");
var UpdatePasswordPage = /* @__PURE__ */ __name(({
  wrapperProps,
  contentProps,
  renderContent,
  formProps,
  title = void 0
}) => {
  var _a, _b;
  const { onSubmit, ...useFormProps } = formProps || {};
  const {
    register,
    watch,
    handleSubmit,
    formState: { errors }
  } = (0, import_react_hook_form6.useForm)({
    ...useFormProps
  });
  const authProvider = (0, import_core22.useActiveAuthProvider)();
  const { mutate: update, isLoading } = (0, import_core23.useUpdatePassword)({
    v3LegacyAuthProviderCompatible: Boolean(authProvider == null ? void 0 : authProvider.isLegacy)
  });
  const translate = (0, import_core23.useTranslate)();
  const PageTitle = title === false ? null : /* @__PURE__ */ React22.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ React22.createElement(
      ThemedTitleV2,
      {
        collapsed: false,
        wrapperStyles: {
          gap: "8px"
        }
      }
    )
  );
  const Content = /* @__PURE__ */ React22.createElement(import_Card5.default, { ...contentProps ?? {} }, /* @__PURE__ */ React22.createElement(import_CardContent5.default, { sx: { p: "32px", "&:last-child": { pb: "32px" } } }, /* @__PURE__ */ React22.createElement(
    import_Typography12.default,
    {
      component: "h1",
      variant: "h5",
      align: "center",
      style: titleStyles,
      color: "primary",
      fontWeight: 700
    },
    translate("pages.updatePassword.title", "Set New Password")
  ), /* @__PURE__ */ React22.createElement(
    import_Box11.default,
    {
      component: "form",
      onSubmit: handleSubmit((data) => {
        if (onSubmit) {
          return onSubmit(data);
        }
        return update(data);
      })
    },
    /* @__PURE__ */ React22.createElement(
      import_TextField5.default,
      {
        ...register("password", {
          required: translate(
            "pages.updatePassword.errors.requiredPassword",
            "Password required"
          )
        }),
        id: "password",
        margin: "normal",
        fullWidth: true,
        name: "password",
        label: translate(
          "pages.updatePassword.fields.password",
          "New Password"
        ),
        helperText: (_a = errors == null ? void 0 : errors.password) == null ? void 0 : _a.message,
        error: !!(errors == null ? void 0 : errors.password),
        type: "password",
        placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF",
        autoComplete: "current-password",
        sx: {
          m: 0
        }
      }
    ),
    /* @__PURE__ */ React22.createElement(
      import_TextField5.default,
      {
        ...register("confirmPassword", {
          required: translate(
            "pages.updatePassword.errors.requiredConfirmPassword",
            "Confirm Password is required"
          ),
          validate: (value) => {
            if (watch("password") !== value) {
              return translate(
                "pages.updatePassword.errors.confirmPasswordNotMatch",
                "Passwords do not match"
              );
            }
            return true;
          }
        }),
        id: "confirmPassword",
        margin: "normal",
        fullWidth: true,
        name: "confirmPassword",
        label: translate(
          "pages.updatePassword.fields.confirmPassword",
          "Confirm New Password"
        ),
        helperText: (_b = errors == null ? void 0 : errors.confirmPassword) == null ? void 0 : _b.message,
        error: !!(errors == null ? void 0 : errors.confirmPassword),
        type: "password",
        placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF",
        autoComplete: "current-confirm-password",
        sx: {
          mb: 0
        }
      }
    ),
    /* @__PURE__ */ React22.createElement(
      import_Button9.default,
      {
        type: "submit",
        fullWidth: true,
        variant: "contained",
        sx: {
          mt: "24px"
        },
        disabled: isLoading
      },
      translate("pages.updatePassword.buttons.submit", "Update")
    )
  )));
  return /* @__PURE__ */ React22.createElement(React22.Fragment, null, /* @__PURE__ */ React22.createElement(import_Box11.default, { component: "div", style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ React22.createElement(
    import_Container5.default,
    {
      component: "main",
      maxWidth: "xs",
      sx: {
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        minHeight: "100dvh",
        padding: "16px",
        width: "100%",
        maxWidth: "400px"
      }
    },
    renderContent ? renderContent(Content, PageTitle) : /* @__PURE__ */ React22.createElement(React22.Fragment, null, PageTitle, Content)
  )));
}, "UpdatePasswordPage");

// src/components/pages/auth/index.tsx
var AuthPage = /* @__PURE__ */ __name((props) => {
  const { type } = props;
  const renderView = /* @__PURE__ */ __name(() => {
    switch (type) {
      case "register":
        return /* @__PURE__ */ import_react20.default.createElement(RegisterPage, { ...props });
      case "forgotPassword":
        return /* @__PURE__ */ import_react20.default.createElement(ForgotPasswordPage, { ...props });
      case "updatePassword":
        return /* @__PURE__ */ import_react20.default.createElement(UpdatePasswordPage, { ...props });
      default:
        return /* @__PURE__ */ import_react20.default.createElement(LoginPage2, { ...props });
    }
  }, "renderView");
  return /* @__PURE__ */ import_react20.default.createElement(import_react20.default.Fragment, null, renderView());
}, "AuthPage");

// src/components/crud/create/index.tsx
var import_react21 = __toESM(require("react"));
var import_core24 = require("@refinedev/core");
var import_Card6 = __toESM(require("@mui/material/Card"));
var import_CardHeader = __toESM(require("@mui/material/CardHeader"));
var import_CardActions = __toESM(require("@mui/material/CardActions"));
var import_CardContent6 = __toESM(require("@mui/material/CardContent"));
var import_IconButton6 = __toESM(require("@mui/material/IconButton"));
var import_Typography13 = __toESM(require("@mui/material/Typography"));
var import_Box12 = __toESM(require("@mui/material/Box"));
var import_ArrowBack = __toESM(require("@mui/icons-material/ArrowBack"));
var import_ui_types = require("@refinedev/ui-types");
var Create = /* @__PURE__ */ __name(({
  title,
  children,
  saveButtonProps: saveButtonPropsFromProps,
  resource: resourceFromProps,
  isLoading = false,
  breadcrumb: breadcrumbFromProps,
  wrapperProps,
  headerProps,
  contentProps,
  headerButtonProps,
  headerButtons,
  footerButtonProps,
  footerButtons,
  goBack: goBackFromProps
}) => {
  var _a, _b;
  const translate = (0, import_core24.useTranslate)();
  const {
    options: { breadcrumb: globalBreadcrumb } = {}
  } = (0, import_core24.useRefineContext)();
  const routerType = (0, import_core24.useRouterType)();
  const back = (0, import_core24.useBack)();
  const { goBack } = (0, import_core24.useNavigation)();
  const getUserFriendlyName = (0, import_core24.useUserFriendlyName)();
  const { resource, action, identifier } = (0, import_core24.useResource)(resourceFromProps);
  const breadcrumb = typeof breadcrumbFromProps === "undefined" ? globalBreadcrumb : breadcrumbFromProps;
  const breadcrumbComponent = typeof breadcrumb !== "undefined" ? /* @__PURE__ */ import_react21.default.createElement(import_react21.default.Fragment, null, breadcrumb) ?? void 0 : /* @__PURE__ */ import_react21.default.createElement(Breadcrumb, null);
  const saveButtonProps = {
    ...isLoading ? { disabled: true } : {},
    ...saveButtonPropsFromProps
  };
  const defaultFooterButtons = /* @__PURE__ */ import_react21.default.createElement(SaveButton, { ...saveButtonProps });
  return /* @__PURE__ */ import_react21.default.createElement(import_Card6.default, { ...wrapperProps ?? {} }, breadcrumbComponent, /* @__PURE__ */ import_react21.default.createElement(
    import_CardHeader.default,
    {
      sx: {
        display: "flex",
        flexWrap: "wrap",
        ".MuiCardHeader-action": {
          margin: 0,
          alignSelf: "center"
        }
      },
      title: title ?? /* @__PURE__ */ import_react21.default.createElement(
        import_Typography13.default,
        {
          variant: "h5",
          className: import_ui_types.RefinePageHeaderClassNames.Title
        },
        translate(
          `${identifier}.titles.create`,
          `Create ${getUserFriendlyName(
            ((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.label) ?? ((_b = resource == null ? void 0 : resource.options) == null ? void 0 : _b.label) ?? (resource == null ? void 0 : resource.label) ?? identifier,
            "singular"
          )}`
        )
      ),
      avatar: typeof goBackFromProps !== "undefined" ? goBackFromProps : /* @__PURE__ */ import_react21.default.createElement(
        import_IconButton6.default,
        {
          onClick: action !== "list" || typeof action !== "undefined" ? routerType === "legacy" ? goBack : back : void 0
        },
        /* @__PURE__ */ import_react21.default.createElement(import_ArrowBack.default, null)
      ),
      action: headerButtons ? /* @__PURE__ */ import_react21.default.createElement(import_Box12.default, { display: "flex", gap: "16px", ...headerButtonProps ?? {} }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
        defaultButtons: null
      }) : headerButtons : null) : void 0,
      ...headerProps ?? {}
    }
  ), /* @__PURE__ */ import_react21.default.createElement(import_CardContent6.default, { ...contentProps ?? {} }, children), /* @__PURE__ */ import_react21.default.createElement(
    import_CardActions.default,
    {
      sx: {
        display: "flex",
        justifyContent: "flex-end",
        gap: "16px",
        padding: "16px"
      },
      ...footerButtonProps ?? {}
    },
    footerButtons ? typeof footerButtons === "function" ? footerButtons({
      defaultButtons: defaultFooterButtons,
      saveButtonProps
    }) : footerButtons : defaultFooterButtons
  ));
}, "Create");

// src/components/crud/edit/index.tsx
var import_react22 = __toESM(require("react"));
var import_core25 = require("@refinedev/core");
var import_Card7 = __toESM(require("@mui/material/Card"));
var import_CardHeader2 = __toESM(require("@mui/material/CardHeader"));
var import_IconButton7 = __toESM(require("@mui/material/IconButton"));
var import_CardContent7 = __toESM(require("@mui/material/CardContent"));
var import_CardActions2 = __toESM(require("@mui/material/CardActions"));
var import_Typography14 = __toESM(require("@mui/material/Typography"));
var import_Box13 = __toESM(require("@mui/material/Box"));
var import_ArrowBack2 = __toESM(require("@mui/icons-material/ArrowBack"));
var import_ui_types2 = require("@refinedev/ui-types");
var Edit = /* @__PURE__ */ __name(({
  title,
  saveButtonProps: saveButtonPropsFromProps,
  mutationMode: mutationModeProp,
  recordItemId,
  children,
  deleteButtonProps: deleteButtonPropsFromProps,
  canDelete,
  resource: resourceFromProps,
  isLoading = false,
  breadcrumb: breadcrumbFromProps,
  dataProviderName,
  wrapperProps,
  headerProps,
  contentProps,
  headerButtonProps,
  headerButtons,
  footerButtonProps,
  footerButtons,
  goBack: goBackFromProps,
  autoSaveProps
}) => {
  var _a, _b, _c;
  const translate = (0, import_core25.useTranslate)();
  const {
    options: { breadcrumb: globalBreadcrumb } = {}
  } = (0, import_core25.useRefineContext)();
  const { mutationMode: mutationModeContext } = (0, import_core25.useMutationMode)();
  const mutationMode = mutationModeProp ?? mutationModeContext;
  const routerType = (0, import_core25.useRouterType)();
  const back = (0, import_core25.useBack)();
  const go = (0, import_core25.useGo)();
  const { goBack, list: legacyGoList } = (0, import_core25.useNavigation)();
  const getUserFriendlyName = (0, import_core25.useUserFriendlyName)();
  const {
    resource,
    action,
    id: idFromParams,
    identifier
  } = (0, import_core25.useResource)(resourceFromProps);
  const goListPath = (0, import_core25.useToPath)({
    resource,
    action: "list"
  });
  const id = recordItemId ?? idFromParams;
  const breadcrumb = typeof breadcrumbFromProps === "undefined" ? globalBreadcrumb : breadcrumbFromProps;
  const hasList = (resource == null ? void 0 : resource.list) && !recordItemId;
  const isDeleteButtonVisible = canDelete ?? ((((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.canDelete) ?? (resource == null ? void 0 : resource.canDelete)) || deleteButtonPropsFromProps);
  const breadcrumbComponent = typeof breadcrumb !== "undefined" ? /* @__PURE__ */ import_react22.default.createElement(import_react22.default.Fragment, null, breadcrumb) ?? void 0 : /* @__PURE__ */ import_react22.default.createElement(Breadcrumb, null);
  const listButtonProps = hasList ? {
    ...isLoading ? { disabled: true } : {},
    resource: routerType === "legacy" ? resource == null ? void 0 : resource.route : identifier
  } : void 0;
  const refreshButtonProps = {
    ...isLoading ? { disabled: true } : {},
    resource: routerType === "legacy" ? resource == null ? void 0 : resource.route : identifier,
    recordItemId: id,
    dataProviderName
  };
  const defaultHeaderButtons = /* @__PURE__ */ import_react22.default.createElement(import_Box13.default, { display: "flex", flexDirection: "row", alignItems: "center" }, autoSaveProps && /* @__PURE__ */ import_react22.default.createElement(AutoSaveIndicator, { ...autoSaveProps }), hasList && /* @__PURE__ */ import_react22.default.createElement(ListButton, { ...listButtonProps }), /* @__PURE__ */ import_react22.default.createElement(RefreshButton, { ...refreshButtonProps }));
  const deleteButtonProps = isDeleteButtonVisible ? {
    ...isLoading ? { disabled: true } : {},
    resource: routerType === "legacy" ? resource == null ? void 0 : resource.route : identifier,
    mutationMode,
    variant: "outlined",
    onSuccess: () => {
      if (routerType === "legacy") {
        legacyGoList((resource == null ? void 0 : resource.route) ?? (resource == null ? void 0 : resource.name) ?? "");
      } else {
        go({ to: goListPath });
      }
    },
    recordItemId: id,
    dataProviderName,
    ...deleteButtonPropsFromProps
  } : void 0;
  const saveButtonProps = {
    ...isLoading ? { disabled: true } : {},
    ...saveButtonPropsFromProps
  };
  const defaultFooterButtons = /* @__PURE__ */ import_react22.default.createElement(import_react22.default.Fragment, null, isDeleteButtonVisible && /* @__PURE__ */ import_react22.default.createElement(DeleteButton, { ...deleteButtonProps }), /* @__PURE__ */ import_react22.default.createElement(SaveButton, { ...saveButtonProps }));
  return /* @__PURE__ */ import_react22.default.createElement(import_Card7.default, { ...wrapperProps ?? {} }, breadcrumbComponent, /* @__PURE__ */ import_react22.default.createElement(
    import_CardHeader2.default,
    {
      sx: {
        display: "flex",
        flexWrap: "wrap",
        ".MuiCardHeader-action": {
          margin: 0,
          alignSelf: "center"
        }
      },
      title: title ?? /* @__PURE__ */ import_react22.default.createElement(
        import_Typography14.default,
        {
          variant: "h5",
          className: import_ui_types2.RefinePageHeaderClassNames.Title
        },
        translate(
          `${identifier}.titles.edit`,
          `Edit ${getUserFriendlyName(
            ((_b = resource == null ? void 0 : resource.meta) == null ? void 0 : _b.label) ?? ((_c = resource == null ? void 0 : resource.options) == null ? void 0 : _c.label) ?? (resource == null ? void 0 : resource.label) ?? identifier,
            "singular"
          )}`
        )
      ),
      avatar: typeof goBackFromProps !== "undefined" ? goBackFromProps : /* @__PURE__ */ import_react22.default.createElement(
        import_IconButton7.default,
        {
          onClick: action !== "list" && typeof action !== "undefined" ? routerType === "legacy" ? goBack : back : void 0
        },
        /* @__PURE__ */ import_react22.default.createElement(import_ArrowBack2.default, null)
      ),
      action: /* @__PURE__ */ import_react22.default.createElement(import_Box13.default, { display: "flex", gap: "16px", ...headerButtonProps ?? {} }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
        defaultButtons: defaultHeaderButtons,
        listButtonProps,
        refreshButtonProps
      }) : headerButtons : defaultHeaderButtons),
      ...headerProps ?? {}
    }
  ), /* @__PURE__ */ import_react22.default.createElement(import_CardContent7.default, { ...contentProps ?? {} }, children), /* @__PURE__ */ import_react22.default.createElement(
    import_CardActions2.default,
    {
      sx: {
        display: "flex",
        justifyContent: "flex-end",
        gap: "16px",
        padding: "16px"
      },
      ...footerButtonProps ?? {}
    },
    footerButtons ? typeof footerButtons === "function" ? footerButtons({
      defaultButtons: defaultFooterButtons,
      deleteButtonProps,
      saveButtonProps
    }) : footerButtons : defaultFooterButtons
  ));
}, "Edit");

// src/components/crud/list/index.tsx
var import_react23 = __toESM(require("react"));
var import_core26 = require("@refinedev/core");
var import_Card8 = __toESM(require("@mui/material/Card"));
var import_CardHeader3 = __toESM(require("@mui/material/CardHeader"));
var import_CardContent8 = __toESM(require("@mui/material/CardContent"));
var import_Typography15 = __toESM(require("@mui/material/Typography"));
var import_Box14 = __toESM(require("@mui/material/Box"));
var import_ui_types3 = require("@refinedev/ui-types");
var List4 = /* @__PURE__ */ __name(({
  title,
  canCreate,
  children,
  createButtonProps: createButtonPropsFromProps,
  resource: resourceFromProps,
  breadcrumb: breadcrumbFromProps,
  wrapperProps,
  headerProps,
  contentProps,
  headerButtonProps,
  headerButtons
}) => {
  var _a, _b;
  const translate = (0, import_core26.useTranslate)();
  const {
    options: { breadcrumb: globalBreadcrumb } = {}
  } = (0, import_core26.useRefineContext)();
  const getUserFriendlyName = (0, import_core26.useUserFriendlyName)();
  const routerType = (0, import_core26.useRouterType)();
  const { resource, identifier } = (0, import_core26.useResource)(resourceFromProps);
  const isCreateButtonVisible = canCreate ?? (((resource == null ? void 0 : resource.canCreate) ?? !!(resource == null ? void 0 : resource.create)) || createButtonPropsFromProps);
  const breadcrumb = typeof breadcrumbFromProps === "undefined" ? globalBreadcrumb : breadcrumbFromProps;
  const breadcrumbComponent = typeof breadcrumb !== "undefined" ? /* @__PURE__ */ import_react23.default.createElement(import_react23.default.Fragment, null, breadcrumb) ?? void 0 : /* @__PURE__ */ import_react23.default.createElement(Breadcrumb, null);
  const createButtonProps = isCreateButtonVisible ? {
    resource: routerType === "legacy" ? resource == null ? void 0 : resource.route : identifier,
    ...createButtonPropsFromProps
  } : void 0;
  const defaultHeaderButtons = isCreateButtonVisible ? /* @__PURE__ */ import_react23.default.createElement(CreateButton, { ...createButtonProps }) : null;
  return /* @__PURE__ */ import_react23.default.createElement(import_Card8.default, { ...wrapperProps ?? {} }, breadcrumbComponent, /* @__PURE__ */ import_react23.default.createElement(
    import_CardHeader3.default,
    {
      sx: {
        display: "flex",
        flexWrap: "wrap",
        ".MuiCardHeader-action": {
          margin: 0,
          alignSelf: "center"
        }
      },
      title: title ?? /* @__PURE__ */ import_react23.default.createElement(
        import_Typography15.default,
        {
          variant: "h5",
          className: import_ui_types3.RefinePageHeaderClassNames.Title
        },
        translate(
          `${identifier}.titles.list`,
          getUserFriendlyName(
            ((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.label) ?? ((_b = resource == null ? void 0 : resource.options) == null ? void 0 : _b.label) ?? (resource == null ? void 0 : resource.label) ?? identifier,
            "plural"
          )
        )
      ),
      action: /* @__PURE__ */ import_react23.default.createElement(import_Box14.default, { display: "flex", gap: "16px", ...headerButtonProps }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
        defaultButtons: defaultHeaderButtons,
        createButtonProps
      }) : headerButtons : defaultHeaderButtons),
      ...headerProps ?? {}
    }
  ), /* @__PURE__ */ import_react23.default.createElement(import_CardContent8.default, { ...contentProps ?? {} }, children));
}, "List");

// src/components/crud/show/index.tsx
var import_react24 = __toESM(require("react"));
var import_core27 = require("@refinedev/core");
var import_Card9 = __toESM(require("@mui/material/Card"));
var import_CardActions3 = __toESM(require("@mui/material/CardActions"));
var import_CardContent9 = __toESM(require("@mui/material/CardContent"));
var import_CardHeader4 = __toESM(require("@mui/material/CardHeader"));
var import_IconButton8 = __toESM(require("@mui/material/IconButton"));
var import_Box15 = __toESM(require("@mui/material/Box"));
var import_Typography16 = __toESM(require("@mui/material/Typography"));
var import_ArrowBack3 = __toESM(require("@mui/icons-material/ArrowBack"));
var import_ui_types4 = require("@refinedev/ui-types");
var Show = /* @__PURE__ */ __name(({
  title,
  canEdit,
  canDelete,
  isLoading = false,
  children,
  resource: resourceFromProps,
  recordItemId,
  breadcrumb: breadcrumbFromProps,
  dataProviderName,
  wrapperProps,
  headerProps,
  contentProps,
  headerButtonProps,
  headerButtons,
  footerButtonProps,
  footerButtons,
  goBack: goBackFromProps
}) => {
  var _a, _b, _c;
  const translate = (0, import_core27.useTranslate)();
  const {
    options: { breadcrumb: globalBreadcrumb } = {}
  } = (0, import_core27.useRefineContext)();
  const routerType = (0, import_core27.useRouterType)();
  const back = (0, import_core27.useBack)();
  const go = (0, import_core27.useGo)();
  const { goBack, list: legacyGoList } = (0, import_core27.useNavigation)();
  const getUserFriendlyName = (0, import_core27.useUserFriendlyName)();
  const {
    resource,
    action,
    id: idFromParams,
    identifier
  } = (0, import_core27.useResource)(resourceFromProps);
  const goListPath = (0, import_core27.useToPath)({
    resource,
    action: "list"
  });
  const id = recordItemId ?? idFromParams;
  const breadcrumb = typeof breadcrumbFromProps === "undefined" ? globalBreadcrumb : breadcrumbFromProps;
  const hasList = (resource == null ? void 0 : resource.list) && !recordItemId;
  const hasDelete = canDelete ?? ((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.canDelete) ?? (resource == null ? void 0 : resource.canDelete);
  const isDeleteButtonVisible = hasDelete && typeof id !== "undefined";
  const isEditButtonVisible = canEdit ?? (resource == null ? void 0 : resource.canEdit) ?? !!(resource == null ? void 0 : resource.edit);
  const breadcrumbComponent = typeof breadcrumb !== "undefined" ? /* @__PURE__ */ import_react24.default.createElement(import_react24.default.Fragment, null, breadcrumb) ?? void 0 : /* @__PURE__ */ import_react24.default.createElement(Breadcrumb, null);
  const listButtonProps = hasList ? {
    ...isLoading ? { disabled: true } : {},
    resource: routerType === "legacy" ? resource == null ? void 0 : resource.route : identifier
  } : void 0;
  const editButtonProps = isEditButtonVisible ? {
    ...isLoading ? { disabled: true } : {},
    resource: routerType === "legacy" ? resource == null ? void 0 : resource.route : identifier,
    recordItemId: id
  } : void 0;
  const deleteButtonProps = isDeleteButtonVisible ? {
    ...isLoading ? { disabled: true } : {},
    resource: routerType === "legacy" ? resource == null ? void 0 : resource.route : identifier,
    recordItemId: id,
    onSuccess: () => {
      if (routerType === "legacy") {
        legacyGoList((resource == null ? void 0 : resource.route) ?? (resource == null ? void 0 : resource.name) ?? "");
      } else {
        go({ to: goListPath });
      }
    },
    dataProviderName
  } : void 0;
  const refreshButtonProps = {
    ...isLoading ? { disabled: true } : {},
    resource: routerType === "legacy" ? resource == null ? void 0 : resource.route : identifier,
    recordItemId: id,
    dataProviderName
  };
  const defaultHeaderButtons = /* @__PURE__ */ import_react24.default.createElement(import_react24.default.Fragment, null, hasList && /* @__PURE__ */ import_react24.default.createElement(ListButton, { ...listButtonProps }), isEditButtonVisible && /* @__PURE__ */ import_react24.default.createElement(EditButton, { ...editButtonProps }), isDeleteButtonVisible && /* @__PURE__ */ import_react24.default.createElement(DeleteButton, { ...deleteButtonProps }), /* @__PURE__ */ import_react24.default.createElement(RefreshButton, { ...refreshButtonProps }));
  return /* @__PURE__ */ import_react24.default.createElement(import_Card9.default, { ...wrapperProps ?? {} }, breadcrumbComponent, /* @__PURE__ */ import_react24.default.createElement(
    import_CardHeader4.default,
    {
      sx: {
        display: "flex",
        flexWrap: "wrap",
        ".MuiCardHeader-action": {
          margin: 0,
          alignSelf: "center"
        }
      },
      title: title ?? /* @__PURE__ */ import_react24.default.createElement(
        import_Typography16.default,
        {
          variant: "h5",
          className: import_ui_types4.RefinePageHeaderClassNames.Title
        },
        translate(
          `${identifier}.titles.show`,
          `Show ${getUserFriendlyName(
            ((_b = resource == null ? void 0 : resource.meta) == null ? void 0 : _b.label) ?? ((_c = resource == null ? void 0 : resource.options) == null ? void 0 : _c.label) ?? (resource == null ? void 0 : resource.label) ?? identifier,
            "singular"
          )}`
        )
      ),
      avatar: typeof goBackFromProps !== "undefined" ? goBackFromProps : /* @__PURE__ */ import_react24.default.createElement(
        import_IconButton8.default,
        {
          onClick: action !== "list" && typeof action !== "undefined" ? routerType === "legacy" ? goBack : back : void 0
        },
        /* @__PURE__ */ import_react24.default.createElement(import_ArrowBack3.default, null)
      ),
      action: /* @__PURE__ */ import_react24.default.createElement(import_Box15.default, { display: "flex", gap: "16px", ...headerButtonProps ?? {} }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
        defaultButtons: defaultHeaderButtons,
        deleteButtonProps,
        editButtonProps,
        listButtonProps,
        refreshButtonProps
      }) : headerButtons : defaultHeaderButtons),
      ...headerProps ?? {}
    }
  ), /* @__PURE__ */ import_react24.default.createElement(import_CardContent9.default, { ...contentProps ?? {} }, children), /* @__PURE__ */ import_react24.default.createElement(import_CardActions3.default, { sx: { padding: "16px" }, ...footerButtonProps ?? {} }, footerButtons ? typeof footerButtons === "function" ? footerButtons({ defaultButtons: null }) : footerButtons : null));
}, "Show");

// src/components/buttons/create/index.tsx
var import_react25 = __toESM(require("react"));
var import_core28 = require("@refinedev/core");
var import_ui_types5 = require("@refinedev/ui-types");
var import_Button10 = __toESM(require("@mui/material/Button"));
var import_AddBoxOutlined = __toESM(require("@mui/icons-material/AddBoxOutlined"));
var CreateButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  resourceNameOrRouteName,
  hideText = false,
  accessControl,
  svgIconProps,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, disabled, hidden, LinkComponent } = (0, import_core28.useCreateButton)(
    {
      resource: resourceNameFromProps ?? resourceNameOrRouteName,
      meta,
      accessControl
    }
  );
  if (hidden)
    return null;
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ import_react25.default.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (disabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      },
      style: { textDecoration: "none" }
    },
    /* @__PURE__ */ import_react25.default.createElement(
      import_Button10.default,
      {
        disabled,
        startIcon: !hideText && /* @__PURE__ */ import_react25.default.createElement(import_AddBoxOutlined.default, { ...svgIconProps }),
        title,
        variant: "contained",
        sx: { minWidth: 0, ...sx },
        
        className: import_ui_types5.RefineButtonClassNames.CreateButton,
        ...restProps
      },
      hideText ? /* @__PURE__ */ import_react25.default.createElement(import_AddBoxOutlined.default, { fontSize: "small", ...svgIconProps }) : children ?? label
    )
  );
}, "CreateButton");

// src/components/buttons/edit/index.tsx
var import_react26 = __toESM(require("react"));
var import_core29 = require("@refinedev/core");
var import_ui_types6 = require("@refinedev/ui-types");
var import_Button11 = __toESM(require("@mui/material/Button"));
var import_EditOutlined = __toESM(require("@mui/icons-material/EditOutlined"));
var EditButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  resourceNameOrRouteName,
  recordItemId,
  hideText = false,
  accessControl,
  svgIconProps,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = (0, import_core29.useEditButton)({
    resource: resourceNameFromProps ?? resourceNameOrRouteName,
    id: recordItemId,
    accessControl,
    meta
  });
  if (hidden)
    return null;
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ import_react26.default.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (disabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      },
      style: { textDecoration: "none" }
    },
    /* @__PURE__ */ import_react26.default.createElement(
      import_Button11.default,
      {
        disabled,
        startIcon: !hideText && /* @__PURE__ */ import_react26.default.createElement(import_EditOutlined.default, { sx: { selfAlign: "center" }, ...svgIconProps }),
        title,
        sx: { minWidth: 0, ...sx },
        
        className: import_ui_types6.RefineButtonClassNames.EditButton,
        ...restProps
      },
      hideText ? /* @__PURE__ */ import_react26.default.createElement(import_EditOutlined.default, { fontSize: "small", ...svgIconProps }) : children ?? label
    )
  );
}, "EditButton");

// src/components/buttons/delete/index.tsx
var import_react27 = __toESM(require("react"));
var import_core30 = require("@refinedev/core");
var import_ui_types7 = require("@refinedev/ui-types");
var import_Button12 = __toESM(require("@mui/material/Button"));
var import_Dialog = __toESM(require("@mui/material/Dialog"));
var import_DialogActions = __toESM(require("@mui/material/DialogActions"));
var import_DialogTitle = __toESM(require("@mui/material/DialogTitle"));
var import_LoadingButton = __toESM(require("@mui/lab/LoadingButton"));
var import_DeleteOutline = __toESM(require("@mui/icons-material/DeleteOutline"));
var DeleteButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  resourceNameOrRouteName,
  recordItemId,
  onSuccess,
  mutationMode,
  children,
  successNotification,
  errorNotification,
  hideText = false,
  accessControl,
  meta,
  metaData,
  dataProviderName,
  confirmTitle,
  confirmOkText,
  confirmCancelText,
  svgIconProps,
  invalidates,
  ...rest
}) => {
  const {
    onConfirm,
    title,
    label,
    hidden,
    disabled,
    loading,
    confirmTitle: defaultConfirmTitle,
    confirmOkLabel,
    cancelLabel
  } = (0, import_core30.useDeleteButton)({
    resource: resourceNameFromProps ?? resourceNameOrRouteName,
    id: recordItemId,
    dataProviderName,
    mutationMode,
    accessControl,
    invalidates,
    onSuccess,
    meta,
    successNotification,
    errorNotification
  });
  const [open, setOpen] = import_react27.default.useState(false);
  const { sx, ...restProps } = rest;
  if (hidden)
    return null;
  return /* @__PURE__ */ import_react27.default.createElement("div", null, /* @__PURE__ */ import_react27.default.createElement(
    import_LoadingButton.default,
    {
      color: "error",
      onClick: () => setOpen(true),
      disabled,
      loading,
      startIcon: !hideText && /* @__PURE__ */ import_react27.default.createElement(import_DeleteOutline.default, { ...svgIconProps }),
      title,
      sx: { minWidth: 0, ...sx },
      loadingPosition: hideText ? "center" : "start",
      
      className: import_ui_types7.RefineButtonClassNames.DeleteButton,
      ...restProps
    },
    hideText ? /* @__PURE__ */ import_react27.default.createElement(import_DeleteOutline.default, { fontSize: "small", ...svgIconProps }) : children ?? label
  ), /* @__PURE__ */ import_react27.default.createElement(
    import_Dialog.default,
    {
      open,
      onClose: () => setOpen(false),
      "aria-labelledby": "alert-dialog-title",
      "aria-describedby": "alert-dialog-description"
    },
    /* @__PURE__ */ import_react27.default.createElement(import_DialogTitle.default, { id: "alert-dialog-title" }, confirmTitle ?? defaultConfirmTitle),
    /* @__PURE__ */ import_react27.default.createElement(import_DialogActions.default, { sx: { justifyContent: "center" } }, /* @__PURE__ */ import_react27.default.createElement(import_Button12.default, { onClick: () => setOpen(false) }, confirmCancelText ?? cancelLabel), /* @__PURE__ */ import_react27.default.createElement(
      import_Button12.default,
      {
        color: "error",
        onClick: () => {
          onConfirm();
          setOpen(false);
        },
        autoFocus: true
      },
      confirmOkText ?? confirmOkLabel
    ))
  ));
}, "DeleteButton");

// src/components/buttons/refresh/index.tsx
var import_react28 = __toESM(require("react"));
var import_core31 = require("@refinedev/core");
var import_ui_types8 = require("@refinedev/ui-types");
var import_LoadingButton2 = __toESM(require("@mui/lab/LoadingButton"));
var import_RefreshOutlined = __toESM(require("@mui/icons-material/RefreshOutlined"));
var RefreshButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  resourceNameOrRouteName,
  recordItemId,
  hideText = false,
  dataProviderName,
  svgIconProps,
  children,
  onClick,
  meta: _meta,
  metaData: _metaData,
  ...rest
}) => {
  const {
    onClick: onRefresh,
    loading,
    label
  } = (0, import_core31.useRefreshButton)({
    resource: resourceNameFromProps ?? resourceNameOrRouteName,
    id: recordItemId,
    dataProviderName
  });
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ import_react28.default.createElement(
    import_LoadingButton2.default,
    {
      startIcon: !hideText && /* @__PURE__ */ import_react28.default.createElement(import_RefreshOutlined.default, { ...svgIconProps }),
      loading,
      loadingPosition: hideText ? "center" : "start",
      onClick: onClick ? onClick : onRefresh,
      sx: { minWidth: 0, ...sx },
      
      className: import_ui_types8.RefineButtonClassNames.RefreshButton,
      ...restProps
    },
    hideText ? /* @__PURE__ */ import_react28.default.createElement(import_RefreshOutlined.default, { fontSize: "small", ...svgIconProps }) : children ?? label
  );
}, "RefreshButton");

// src/components/buttons/show/index.tsx
var import_react29 = __toESM(require("react"));
var import_core32 = require("@refinedev/core");
var import_ui_types9 = require("@refinedev/ui-types");
var import_Button13 = __toESM(require("@mui/material/Button"));
var import_VisibilityOutlined = __toESM(require("@mui/icons-material/VisibilityOutlined"));
var ShowButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  resourceNameOrRouteName,
  recordItemId,
  hideText = false,
  accessControl,
  svgIconProps,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = (0, import_core32.useShowButton)({
    resource: resourceNameFromProps ?? resourceNameOrRouteName,
    id: recordItemId,
    accessControl,
    meta
  });
  if (hidden)
    return null;
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ import_react29.default.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (disabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      },
      style: { textDecoration: "none" }
    },
    /* @__PURE__ */ import_react29.default.createElement(
      import_Button13.default,
      {
        disabled,
        startIcon: !hideText && /* @__PURE__ */ import_react29.default.createElement(import_VisibilityOutlined.default, { ...svgIconProps }),
        title,
        sx: { minWidth: 0, ...sx },
        
        className: import_ui_types9.RefineButtonClassNames.ShowButton,
        ...restProps
      },
      hideText ? /* @__PURE__ */ import_react29.default.createElement(import_VisibilityOutlined.default, { fontSize: "small", ...svgIconProps }) : children ?? label
    )
  );
}, "ShowButton");

// src/components/buttons/list/index.tsx
var import_react30 = __toESM(require("react"));
var import_core33 = require("@refinedev/core");
var import_ui_types10 = require("@refinedev/ui-types");
var import_Button14 = __toESM(require("@mui/material/Button"));
var import_ListOutlined4 = __toESM(require("@mui/icons-material/ListOutlined"));
var ListButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  resourceNameOrRouteName,
  hideText = false,
  accessControl,
  svgIconProps,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = (0, import_core33.useListButton)({
    resource: resourceNameFromProps ?? resourceNameOrRouteName,
    meta,
    accessControl
  });
  if (hidden)
    return null;
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ import_react30.default.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (disabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      },
      style: { textDecoration: "none" }
    },
    /* @__PURE__ */ import_react30.default.createElement(
      import_Button14.default,
      {
        disabled,
        startIcon: !hideText && /* @__PURE__ */ import_react30.default.createElement(import_ListOutlined4.default, { ...svgIconProps }),
        title,
        sx: { minWidth: 0, ...sx },
        
        className: import_ui_types10.RefineButtonClassNames.ListButton,
        ...restProps
      },
      hideText ? /* @__PURE__ */ import_react30.default.createElement(import_ListOutlined4.default, { fontSize: "small", ...svgIconProps }) : children ?? label
    )
  );
}, "ListButton");

// src/components/buttons/export/index.tsx
var import_react31 = __toESM(require("react"));
var import_core34 = require("@refinedev/core");
var import_ui_types11 = require("@refinedev/ui-types");
var import_LoadingButton3 = __toESM(require("@mui/lab/LoadingButton"));
var import_ImportExportOutlined = __toESM(require("@mui/icons-material/ImportExportOutlined"));
var ExportButton = /* @__PURE__ */ __name(({
  hideText = false,
  children,
  loading = false,
  svgIconProps,
  ...rest
}) => {
  const { label } = (0, import_core34.useExportButton)();
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ import_react31.default.createElement(
    import_LoadingButton3.default,
    {
      ...rest,
      loading,
      startIcon: !hideText && /* @__PURE__ */ import_react31.default.createElement(import_ImportExportOutlined.default, { ...svgIconProps }),
      loadingPosition: hideText ? "center" : "start",
      sx: { minWidth: 0, ...sx },
      
      className: import_ui_types11.RefineButtonClassNames.ExportButton,
      ...restProps
    },
    hideText ? /* @__PURE__ */ import_react31.default.createElement(import_ImportExportOutlined.default, { fontSize: "small", ...svgIconProps }) : children ?? label
  );
}, "ExportButton");

// src/components/buttons/save/index.tsx
var import_react32 = __toESM(require("react"));
var import_core35 = require("@refinedev/core");
var import_ui_types12 = require("@refinedev/ui-types");
var import_LoadingButton4 = __toESM(require("@mui/lab/LoadingButton"));
var import_SaveOutlined = __toESM(require("@mui/icons-material/SaveOutlined"));
var SaveButton = /* @__PURE__ */ __name(({
  hideText = false,
  svgIconProps,
  children,
  ...rest
}) => {
  const { label } = (0, import_core35.useSaveButton)();
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ import_react32.default.createElement(
    import_LoadingButton4.default,
    {
      startIcon: !hideText && /* @__PURE__ */ import_react32.default.createElement(import_SaveOutlined.default, { ...svgIconProps }),
      sx: { minWidth: 0, ...sx },
      variant: "contained",
      
      className: import_ui_types12.RefineButtonClassNames.SaveButton,
      ...restProps
    },
    hideText ? /* @__PURE__ */ import_react32.default.createElement(import_SaveOutlined.default, { fontSize: "small", ...svgIconProps }) : children ?? label
  );
}, "SaveButton");

// src/components/buttons/clone/index.tsx
var import_react33 = __toESM(require("react"));
var import_core36 = require("@refinedev/core");
var import_ui_types13 = require("@refinedev/ui-types");
var import_Button15 = __toESM(require("@mui/material/Button"));
var import_AddBoxOutlined2 = __toESM(require("@mui/icons-material/AddBoxOutlined"));
var CloneButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  resourceNameOrRouteName,
  recordItemId,
  hideText = false,
  accessControl,
  svgIconProps,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = (0, import_core36.useCloneButton)({
    resource: resourceNameFromProps ?? resourceNameOrRouteName,
    id: recordItemId,
    meta,
    accessControl
  });
  if (hidden)
    return null;
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ import_react33.default.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (disabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      },
      style: { textDecoration: "none" }
    },
    /* @__PURE__ */ import_react33.default.createElement(
      import_Button15.default,
      {
        disabled,
        startIcon: !hideText && /* @__PURE__ */ import_react33.default.createElement(import_AddBoxOutlined2.default, { ...svgIconProps }),
        title,
        sx: { minWidth: 0, ...sx },
        
        className: import_ui_types13.RefineButtonClassNames.CloneButton,
        ...restProps
      },
      hideText ? /* @__PURE__ */ import_react33.default.createElement(import_AddBoxOutlined2.default, { fontSize: "small", ...svgIconProps }) : children ?? label
    )
  );
}, "CloneButton");

// src/components/buttons/import/index.tsx
var import_react34 = __toESM(require("react"));
var import_core37 = require("@refinedev/core");
var import_ui_types14 = require("@refinedev/ui-types");
var import_LoadingButton5 = __toESM(require("@mui/lab/LoadingButton"));
var import_ImportExportOutlined2 = __toESM(require("@mui/icons-material/ImportExportOutlined"));
var ImportButton = /* @__PURE__ */ __name(({
  inputProps,
  hideText = false,
  loading = false,
  svgIconProps,
  children,
  ...rest
}) => {
  const { label } = (0, import_core37.useImportButton)();
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ import_react34.default.createElement("label", { htmlFor: "contained-button-file" }, /* @__PURE__ */ import_react34.default.createElement("input", { ...inputProps, id: "contained-button-file", multiple: true, hidden: true }), /* @__PURE__ */ import_react34.default.createElement(
    import_LoadingButton5.default,
    {
      component: "span",
      startIcon: !hideText && /* @__PURE__ */ import_react34.default.createElement(import_ImportExportOutlined2.default, { ...svgIconProps }),
      loadingPosition: hideText ? "center" : "start",
      loading,
      sx: { minWidth: 0, ...sx },
      
      className: import_ui_types14.RefineButtonClassNames.ImportButton,
      ...restProps
    },
    hideText ? /* @__PURE__ */ import_react34.default.createElement(import_ImportExportOutlined2.default, { fontSize: "small", ...svgIconProps }) : children ?? label
  ));
}, "ImportButton");

// src/components/circularDeterminate/index.tsx
var import_react35 = __toESM(require("react"));
var import_Box16 = __toESM(require("@mui/material/Box"));
var import_CircularProgress = __toESM(require("@mui/material/CircularProgress"));
var import_Typography17 = __toESM(require("@mui/material/Typography"));
var CircularDeterminate = /* @__PURE__ */ __name(({
  undoableTimeout,
  message
}) => {
  const [progress, setProgress] = (0, import_react35.useState)(100);
  const [timeCount, setTimeCount] = (0, import_react35.useState)(undoableTimeout);
  (0, import_react35.useEffect)(() => {
    const increaseProgress = 100 / undoableTimeout;
    const timer = setInterval(() => {
      setTimeCount((prevProgress) => prevProgress - 1);
      setProgress((prevProgress) => prevProgress - increaseProgress);
    }, 1e3);
    if (timeCount === 0) {
      clearInterval(timer);
    }
    return () => {
      clearInterval(timer);
    };
  }, [timeCount]);
  return /* @__PURE__ */ import_react35.default.createElement(import_react35.default.Fragment, null, /* @__PURE__ */ import_react35.default.createElement(import_Box16.default, { sx: { position: "relative", display: "inline-flex" } }, /* @__PURE__ */ import_react35.default.createElement(
    import_CircularProgress.default,
    {
      color: "inherit",
      variant: "determinate",
      value: progress
    }
  ), /* @__PURE__ */ import_react35.default.createElement(
    import_Box16.default,
    {
      sx: {
        top: 0,
        left: 0,
        bottom: 0,
        right: 0,
        position: "absolute",
        display: "flex",
        alignItems: "center",
        justifyContent: "center"
      }
    },
    /* @__PURE__ */ import_react35.default.createElement(import_Typography17.default, { component: "div" }, timeCount)
  )), /* @__PURE__ */ import_react35.default.createElement(
    import_Box16.default,
    {
      sx: {
        marginLeft: "10px",
        maxWidth: { xs: "150px", md: "100%" }
      }
    },
    /* @__PURE__ */ import_react35.default.createElement(import_Typography17.default, { variant: "subtitle2" }, message)
  ));
}, "CircularDeterminate");

// src/components/fields/text/index.tsx
var import_react36 = __toESM(require("react"));
var import_Typography18 = __toESM(require("@mui/material/Typography"));
var TextField6 = /* @__PURE__ */ __name(({ value, ...rest }) => {
  return /* @__PURE__ */ import_react36.default.createElement(import_Typography18.default, { variant: "body2", ...rest }, value);
}, "TextField");

// src/components/fields/tag/index.tsx
var import_react37 = __toESM(require("react"));
var import_Chip = __toESM(require("@mui/material/Chip"));
var TagField = /* @__PURE__ */ __name(({ value, ...rest }) => {
  return /* @__PURE__ */ import_react37.default.createElement(import_Chip.default, { label: value == null ? void 0 : value.toString(), ...rest });
}, "TagField");

// src/components/fields/email/index.tsx
var import_react38 = __toESM(require("react"));
var import_Typography19 = __toESM(require("@mui/material/Typography"));
var import_Link8 = __toESM(require("@mui/material/Link"));
var EmailField = /* @__PURE__ */ __name(({ value, ...rest }) => {
  return /* @__PURE__ */ import_react38.default.createElement(import_Typography19.default, { variant: "body2" }, /* @__PURE__ */ import_react38.default.createElement(import_Link8.default, { href: `mailto:${value}`, ...rest }, value));
}, "EmailField");

// src/components/fields/boolean/index.tsx
var import_react39 = __toESM(require("react"));
var import_Tooltip5 = __toESM(require("@mui/material/Tooltip"));
var import_CheckOutlined = __toESM(require("@mui/icons-material/CheckOutlined"));
var import_CloseOutlined = __toESM(require("@mui/icons-material/CloseOutlined"));
var BooleanField = /* @__PURE__ */ __name(({
  value,
  valueLabelTrue = "true",
  valueLabelFalse = "false",
  trueIcon,
  falseIcon,
  svgIconProps,
  ...rest
}) => {
  return /* @__PURE__ */ import_react39.default.createElement(import_Tooltip5.default, { title: value ? valueLabelTrue : valueLabelFalse, ...rest }, value ? /* @__PURE__ */ import_react39.default.createElement("span", null, trueIcon ?? /* @__PURE__ */ import_react39.default.createElement(import_CheckOutlined.default, { ...svgIconProps })) : /* @__PURE__ */ import_react39.default.createElement("span", null, falseIcon ?? /* @__PURE__ */ import_react39.default.createElement(import_CloseOutlined.default, { ...svgIconProps })));
}, "BooleanField");

// src/components/fields/date/index.tsx
var import_react40 = __toESM(require("react"));
var import_dayjs = __toESM(require("dayjs"));
var import_localizedFormat = __toESM(require("dayjs/plugin/localizedFormat"));
var import_Typography20 = __toESM(require("@mui/material/Typography"));
import_dayjs.default.extend(import_localizedFormat.default);
var defaultLocale = import_dayjs.default.locale();
var DateField = /* @__PURE__ */ __name(({
  value,
  locales,
  format: dateFormat = "L",
  ...rest
}) => {
  return /* @__PURE__ */ import_react40.default.createElement(import_Typography20.default, { variant: "body2", ...rest }, value ? (0, import_dayjs.default)(value).locale(locales || defaultLocale).format(dateFormat) : "");
}, "DateField");

// src/components/fields/file/index.tsx
var import_react41 = __toESM(require("react"));
var FileField = /* @__PURE__ */ __name(({
  title,
  src,
  ...rest
}) => {
  return /* @__PURE__ */ import_react41.default.createElement(UrlField, { value: src, title, ...rest }, title ?? src);
}, "FileField");

// src/components/fields/url/index.tsx
var import_react42 = __toESM(require("react"));
var import_Link9 = __toESM(require("@mui/material/Link"));
var import_Typography21 = __toESM(require("@mui/material/Typography"));
var UrlField = /* @__PURE__ */ __name(({
  children,
  value,
  ...rest
}) => {
  return /* @__PURE__ */ import_react42.default.createElement(import_Typography21.default, { variant: "body2" }, /* @__PURE__ */ import_react42.default.createElement(import_Link9.default, { href: value, ...rest }, children ?? value));
}, "UrlField");

// src/components/fields/number/index.tsx
var import_react43 = __toESM(require("react"));
var import_Typography22 = __toESM(require("@mui/material/Typography"));
function toLocaleStringSupportsOptions() {
  return !!(typeof Intl === "object" && Intl && typeof Intl.NumberFormat === "function");
}
__name(toLocaleStringSupportsOptions, "toLocaleStringSupportsOptions");
var NumberField = /* @__PURE__ */ __name(({
  value,
  locale,
  options,
  ...rest
}) => {
  const number = Number(value);
  return /* @__PURE__ */ import_react43.default.createElement(import_Typography22.default, { variant: "body2", ...rest }, toLocaleStringSupportsOptions() ? number.toLocaleString(locale, options) : number);
}, "NumberField");

// src/components/fields/markdown/index.tsx
var import_react44 = __toESM(require("react"));
var import_react_markdown = __toESM(require("react-markdown"));
var import_remark_gfm = __toESM(require("remark-gfm"));
var MarkdownField = /* @__PURE__ */ __name(({ value = "" }) => {
  return /* @__PURE__ */ import_react44.default.createElement(
    import_react_markdown.default,
    {
      remarkPlugins: [import_remark_gfm.default]
    },
    value
  );
}, "MarkdownField");

// src/components/breadcrumb/index.tsx
var import_react45 = __toESM(require("react"));
var import_core38 = require("@refinedev/core");
var import_Breadcrumbs = __toESM(require("@mui/material/Breadcrumbs"));
var import_Typography23 = __toESM(require("@mui/material/Typography"));
var import_Link10 = __toESM(require("@mui/material/Link"));
var import_Grid3 = __toESM(require("@mui/material/Grid"));
var import_HomeOutlined = __toESM(require("@mui/icons-material/HomeOutlined"));
var Breadcrumb = /* @__PURE__ */ __name(({
  breadcrumbProps,
  showHome = true,
  hideIcons = false,
  meta
}) => {
  var _a, _b;
  const { breadcrumbs } = (0, import_core38.useBreadcrumb)({ meta });
  const routerType = (0, import_core38.useRouterType)();
  const NewLink = (0, import_core38.useLink)();
  const { Link: LegacyLink } = (0, import_core38.useRouterContext)();
  const ActiveLink = routerType === "legacy" ? LegacyLink : NewLink;
  const { hasDashboard } = (0, import_core38.useRefineContext)();
  const { resources } = (0, import_core38.useResource)();
  const rootRouteResource = (0, import_core38.matchResourceFromRoute)("/", resources);
  if (breadcrumbs.length === 1) {
    return null;
  }
  const LinkRouter = /* @__PURE__ */ __name((props) => /* @__PURE__ */ import_react45.default.createElement(import_Link10.default, { ...props, component: ActiveLink }), "LinkRouter");
  return /* @__PURE__ */ import_react45.default.createElement(
    import_Breadcrumbs.default,
    {
      "aria-label": "breadcrumb",
      sx: {
        paddingY: 2,
        paddingX: 2,
        ...(breadcrumbProps == null ? void 0 : breadcrumbProps.sx) ?? {}
      },
      ...breadcrumbProps
    },
    showHome && (hasDashboard || rootRouteResource.found) && /* @__PURE__ */ import_react45.default.createElement(
      LinkRouter,
      {
        underline: "hover",
        sx: {
          display: "flex",
          alignItems: "center"
        },
        color: "inherit",
        to: "/"
      },
      ((_b = (_a = rootRouteResource == null ? void 0 : rootRouteResource.resource) == null ? void 0 : _a.meta) == null ? void 0 : _b.icon) ?? /* @__PURE__ */ import_react45.default.createElement(
        import_HomeOutlined.default,
        {
          sx: {
            fontSize: "18px"
          }
        }
      )
    ),
    breadcrumbs.map(({ label, icon, href }) => {
      return /* @__PURE__ */ import_react45.default.createElement(
        import_Grid3.default,
        {
          key: label,
          sx: {
            display: "flex",
            alignItems: "center",
            "& .MuiSvgIcon-root": {
              fontSize: "16px"
            }
          }
        },
        !hideIcons && icon,
        href ? /* @__PURE__ */ import_react45.default.createElement(
          LinkRouter,
          {
            underline: "hover",
            sx: {
              display: "flex",
              alignItems: "center",
              fontSize: "14px"
            },
            color: "inherit",
            to: href,
            variant: "subtitle1",
            marginLeft: 0.5
          },
          label
        ) : /* @__PURE__ */ import_react45.default.createElement(import_Typography23.default, { fontSize: "14px" }, label)
      );
    })
  );
}, "Breadcrumb");

// src/components/autoSaveIndicator/index.tsx
var import_react46 = __toESM(require("react"));
var import_core39 = require("@refinedev/core");
var import_Typography24 = __toESM(require("@mui/material/Typography"));
var import_MoreHorizOutlined = __toESM(require("@mui/icons-material/MoreHorizOutlined"));
var import_SyncOutlined = __toESM(require("@mui/icons-material/SyncOutlined"));
var import_TaskAltOutlined = __toESM(require("@mui/icons-material/TaskAltOutlined"));
var import_ErrorOutlineOutlined = __toESM(require("@mui/icons-material/ErrorOutlineOutlined"));
var AutoSaveIndicator = /* @__PURE__ */ __name(({
  status,
  elements: {
    success = /* @__PURE__ */ import_react46.default.createElement(
      Message,
      {
        translationKey: "autoSave.success",
        defaultMessage: "saved",
        icon: /* @__PURE__ */ import_react46.default.createElement(import_TaskAltOutlined.default, { fontSize: "small" })
      }
    ),
    error = /* @__PURE__ */ import_react46.default.createElement(
      Message,
      {
        translationKey: "autoSave.error",
        defaultMessage: "auto save failure",
        icon: /* @__PURE__ */ import_react46.default.createElement(import_ErrorOutlineOutlined.default, { fontSize: "small" })
      }
    ),
    loading = /* @__PURE__ */ import_react46.default.createElement(
      Message,
      {
        translationKey: "autoSave.loading",
        defaultMessage: "saving...",
        icon: /* @__PURE__ */ import_react46.default.createElement(import_SyncOutlined.default, { fontSize: "small" })
      }
    ),
    idle = /* @__PURE__ */ import_react46.default.createElement(
      Message,
      {
        translationKey: "autoSave.idle",
        defaultMessage: "waiting for changes",
        icon: /* @__PURE__ */ import_react46.default.createElement(import_MoreHorizOutlined.default, { fontSize: "small" })
      }
    )
  } = {}
}) => {
  return /* @__PURE__ */ import_react46.default.createElement(
    import_core39.AutoSaveIndicator,
    {
      status,
      elements: {
        success,
        error,
        loading,
        idle
      }
    }
  );
}, "AutoSaveIndicator");
var Message = /* @__PURE__ */ __name(({
  translationKey,
  defaultMessage,
  icon
}) => {
  const translate = (0, import_core39.useTranslate)();
  return /* @__PURE__ */ import_react46.default.createElement(
    import_Typography24.default,
    {
      color: "gray",
      fontSize: "0.8rem",
      position: "relative",
      display: "flex",
      alignItems: "center",
      flexWrap: "wrap",
      marginRight: ".3rem"
    },
    translate(translationKey, defaultMessage),
    /* @__PURE__ */ import_react46.default.createElement("span", { style: { position: "relative", top: "3px", marginLeft: "3px" } }, icon)
  );
}, "Message");

// src/theme/index.ts
var import_styles6 = require("@mui/material/styles");

// src/theme/palette/lightPalette.ts
var lightPalette = {
  mode: "light",
  primary: {
    main: "#67be23",
    contrastText: "#fff"
  },
  secondary: {
    main: "#2A132E",
    contrastText: "#fff"
  },
  background: {
    default: "#f0f0f0",
    paper: "#ffffff"
  },
  success: {
    main: "#67be23",
    contrastText: "#fff"
  },
  error: {
    main: "#fa541c",
    contrastText: "#fff"
  },
  warning: {
    main: "#fa8c16",
    contrastText: "#fff"
  },
  info: {
    main: "#0b82f0",
    contrastText: "#fff"
  },
  divider: "rgba(0,0,0,0)",
  text: {
    primary: "#626262",
    secondary: "#9f9f9f",
    disabled: "#c1c1c1"
  }
};

// src/theme/palette/darkPalette.ts
var darkPalette = {
  mode: "dark",
  primary: {
    main: "#67be23",
    contrastText: "#fff"
  },
  secondary: {
    main: "#2A132E",
    contrastText: "#fff"
  },
  background: {
    default: "#212121",
    paper: "#242424"
  },
  success: {
    main: "#67be23",
    contrastText: "#fff"
  },
  error: {
    main: "#ee2a1e",
    contrastText: "#fff"
  },
  warning: {
    main: "#fa8c16",
    contrastText: "#fff"
  },
  info: {
    main: "#1890ff",
    contrastText: "#fff"
  },
  divider: "rgba(0,0,0,0)",
  text: {
    primary: "#fff",
    secondary: "rgba(255,255,255,0.7)",
    disabled: "#d1d1d1"
  }
};

// src/theme/typography.ts
var typography = {
  fontFamily: [
    "Montserrat",
    "-apple-system",
    "BlinkMacSystemFont",
    '"Segoe UI"',
    "Roboto",
    '"Helvetica Neue"',
    "Arial",
    "sans-serif",
    '"Apple Color Emoji"',
    '"Segoe UI Emoji"',
    '"Segoe UI Symbol"'
  ].join(",")
};

// src/theme/palette/refinePalette.ts
var RefinePalettes = {
  Blue: {
    mode: "light",
    primary: {
      main: "#1976D2",
      light: "#4791db",
      dark: "#115293"
    }
  },
  BlueDark: {
    mode: "dark",
    primary: {
      main: "#67b7f7",
      light: "#85c5f8",
      dark: "#4880ac"
    }
  },
  Purple: {
    mode: "light",
    primary: {
      main: "#7B1FA2",
      light: "#954bb4",
      dark: "#561571"
    }
  },
  PurpleDark: {
    mode: "dark",
    primary: {
      main: "#AB47BC",
      light: "#bb6bc9",
      dark: "#773183"
    }
  },
  Magenta: {
    mode: "light",
    primary: {
      main: "#C2185B",
      light: "#ce467b",
      dark: "#87103f"
    }
  },
  MagentaDark: {
    mode: "dark",
    primary: {
      main: "#EC407A",
      light: "#ef6694",
      dark: "#a52c55"
    }
  },
  Red: {
    mode: "light",
    primary: {
      main: "#D32F2F",
      light: "#db5858",
      dark: "#932020"
    }
  },
  RedDark: {
    mode: "dark",
    primary: {
      main: "#EF5350",
      light: "#f27573",
      dark: "#a73a38"
    }
  },
  Orange: {
    mode: "light",
    primary: {
      main: "#F57C00",
      light: "#f79633",
      dark: "#ab5600"
    }
  },
  OrangeDark: {
    mode: "dark",
    primary: {
      main: "#FFA726",
      light: "#ffb851",
      dark: "#b2741a"
    }
  },
  Yellow: {
    mode: "light",
    primary: {
      main: "#FFA000",
      light: "#ffb333",
      dark: "#b27000"
    }
  },
  YellowDark: {
    mode: "dark",
    primary: {
      main: "#FFCA28",
      light: "#ffd453",
      dark: "#E87040"
    }
  },
  Green: {
    mode: "light",
    primary: {
      main: "#689F38",
      light: "#86b25f",
      dark: "#486f27"
    }
  },
  GreenDark: {
    mode: "dark",
    primary: {
      main: "#9CCC65",
      light: "#afd683",
      dark: "#6d8e46"
    }
  }
};

// src/theme/index.ts
var commonThemeProperties = {
  shape: {
    borderRadius: 6
  },
  typography: {
    ...typography
  }
};
var LightTheme = (0, import_styles6.createTheme)({
  ...commonThemeProperties,
  palette: lightPalette,
  components: {
    MuiAppBar: {
      styleOverrides: {
        colorDefault: {
          backgroundColor: "#fff"
        }
      }
    },
    MuiPaper: {
      styleOverrides: {
        root: {
          backgroundImage: "linear-gradient(rgba(255, 255, 255, 0.01), rgba(255, 255, 255, 0.01))"
        }
      }
    },
    MuiTypography: {
      styleOverrides: {
        h5: {
          fontWeight: 800,
          lineHeight: "2rem"
        }
      }
    }
  }
});
var DarkTheme = (0, import_styles6.createTheme)({
  ...commonThemeProperties,
  palette: darkPalette,
  components: {
    MuiPaper: {
      styleOverrides: {
        root: {
          backgroundImage: "linear-gradient(rgba(255, 255, 255, 0.025), rgba(255, 255, 255, 0.025))"
        }
      }
    },
    MuiAppBar: {
      defaultProps: {
        color: "transparent"
      }
    },
    MuiTypography: {
      styleOverrides: {
        h5: {
          fontWeight: 800,
          lineHeight: "2rem"
        }
      }
    }
  }
});
var RefineThemes = Object.keys(RefinePalettes).reduce((acc, key) => {
  const paletteName = key;
  return {
    ...acc,
    [key]: (0, import_styles6.createTheme)({
      palette: {
        ...RefinePalettes[paletteName]
      },
      components: {
        MuiButton: {
          styleOverrides: {
            root: ({ ownerState }) => ({
              ...ownerState.variant === "contained" && ownerState.color === "primary" && {
                color: "#fff"
              }
            })
          }
        }
      }
    })
  };
}, {});

// src/providers/notificationProvider/index.tsx
var import_react47 = __toESM(require("react"));
var import_notistack = require("notistack");
var import_Box17 = __toESM(require("@mui/material/Box"));
var import_IconButton9 = __toESM(require("@mui/material/IconButton"));
var import_Typography25 = __toESM(require("@mui/material/Typography"));
var import_UndoOutlined = __toESM(require("@mui/icons-material/UndoOutlined"));
var useNotificationProvider = /* @__PURE__ */ __name(() => {
  const { closeSnackbar, enqueueSnackbar } = (0, import_notistack.useSnackbar)();
  const notificationProvider2 = {
    open: ({
      message,
      type,
      undoableTimeout,
      key,
      cancelMutation,
      description
    }) => {
      if (type === "progress") {
        const action = /* @__PURE__ */ __name((key2) => /* @__PURE__ */ import_react47.default.createElement(
          import_IconButton9.default,
          {
            onClick: () => {
              cancelMutation == null ? void 0 : cancelMutation();
              closeSnackbar(key2);
            },
            color: "inherit"
          },
          /* @__PURE__ */ import_react47.default.createElement(import_UndoOutlined.default, null)
        ), "action");
        enqueueSnackbar(
          /* @__PURE__ */ import_react47.default.createElement(import_react47.default.Fragment, null, /* @__PURE__ */ import_react47.default.createElement(
            CircularDeterminate,
            {
              undoableTimeout: undoableTimeout ?? 0,
              message
            }
          )),
          {
            action,
            preventDuplicate: true,
            key,
            autoHideDuration: (undoableTimeout ?? 0) * 1e3
          }
        );
      } else {
        enqueueSnackbar(
          /* @__PURE__ */ import_react47.default.createElement(import_Box17.default, null, /* @__PURE__ */ import_react47.default.createElement(import_Typography25.default, { variant: "subtitle2", component: "h6" }, description), /* @__PURE__ */ import_react47.default.createElement(import_Typography25.default, { variant: "caption", component: "p" }, message)),
          {
            key,
            variant: type
          }
        );
      }
    },
    close: (key) => {
      closeSnackbar(key);
    }
  };
  return notificationProvider2;
}, "useNotificationProvider");
var notificationProvider = useNotificationProvider;

// src/providers/refineSnackbarProvider/index.tsx
var import_react48 = __toESM(require("react"));
var import_styles7 = require("@mui/material/styles");
var import_notistack2 = require("notistack");
var SnackbarProviderWithDefaultValues = /* @__PURE__ */ __name(({
  anchorOrigin = {
    vertical: "top",
    horizontal: "right"
  },
  disableWindowBlurListener = true,
  ...rest
}) => {
  return /* @__PURE__ */ import_react48.default.createElement(
    import_notistack2.SnackbarProvider,
    {
      anchorOrigin,
      disableWindowBlurListener,
      ...rest
    }
  );
}, "SnackbarProviderWithDefaultValues");
var RefineSnackbarProvider = (0, import_styles7.styled)(SnackbarProviderWithDefaultValues)`
&.SnackbarItem-contentRoot {
    background-color: ${(props) => props.theme.palette.background.default};
    color: ${(props) => props.theme.palette.primary.main};
}
&.SnackbarItem-variantSuccess {
    background-color: ${(props) => props.theme.palette.success.main};
    color: ${(props) => props.theme.palette.success.contrastText};
}
&.SnackbarItem-variantError {
    background-color: ${(props) => props.theme.palette.error.main};
    color: ${(props) => props.theme.palette.error.contrastText};
}
&.SnackbarItem-variantInfo {
    background-color: ${(props) => props.theme.palette.info.main};
    color: ${(props) => props.theme.palette.info.contrastText};
}
&.SnackbarItem-variantWarning {
    background-color: ${(props) => props.theme.palette.warning.main};
    color: ${(props) => props.theme.palette.warning.contrastText};
}
`;
